---
title: Describing *Symbiodinium* metacommunities using high-throughput sequencing
  of ITS2
author:
- affiliation: University of Hawaii
  email: ross.cunning@gmail.com
  footnote: Corresponding Author
  name: Ross Cunning
- affiliation: University of Hawaii
  name: Ruth D. Gates
- affiliation: California State University Northridge
  name: Peter J. Edmunds
bibliography: library.bib
output:
  word_document: default
  pdf_document: default
  html_document:
    fig_caption: yes
subtitle: A case study from St. John, U.S. Virgin Islands
address:
- address: Hawaii Institute of Marine Biology, Kaneohe, HI 96744, USA
  code: University of Hawaii
- address: Department of Biology, Northridge, CA 91330, USA
  code: California State University Northridge
---
```{r setup, include=FALSE}
knitr::opts_knit$set(root.dir = normalizePath(".."))
knitr::opts_chunk$set(echo = F, warning = F, message = F, cache = T)
options(scipen = 3, digits = 9)
```

```{r setup_2, cache=F, include=FALSE}
# Load all R scripts in "R/" directory
source("R/functions.R", .GlobalEnv)
# Load package libraries
library(phyloseq); library(vegan); library(multcompView); library(reshape2); library(igraph); library(stringr); library(RColorBrewer); library(pBrackets)
# Set colors for plotting clades
taxcolors <- matrix(c("#8dd3c7", "#ffffb3", "#bebada", "#fb8072", "#80b1d3", "#fdb462"), 
                    dimnames=list(c("CladeA", "CladeB", "CladeC", "CladeD", "CladeF", "CladeG")))
```

```{r import_taxa_data, include=F}
# Import taxonomic assignment data from nw
read.nw <- function(file) {
  nw <- read.table(file, stringsAsFactors=FALSE)
  nw <- nw[order(nw$otu),]
  return(nw)
}

tax97 <- read.nw("data/otus_97/nw_tophits.tsv")
tax100 <- read.nw("data/otus_100/nw_tophits.tsv")
tax97bs <- read.nw("data/otus_97_bysample/nw_tophits.tsv")

# Deal with identical taxonomic assignments (because some reference sequences are not unique...)
# Create names for groups of identical sequences based on members of group...
ident <- readLines("data/ITS2db_trimmed_notuniques_otus.txt")
ident <- gsub("denovo[0-9]*\t", "", ident)
ident <- strsplit(ident, split="\t")
ident2 <- lapply(ident, str_match_all, pattern="[A-I]{1}[0-9]{1,3}.*[_/]")
ident2 <- lapply(ident2, function(g) gsub(pattern="\\..*_$", "", x=unlist(g)))
ident2 <- lapply(ident2, function(g) gsub(pattern="_$", "", g))
ident2 <- lapply(ident2, function(g) unlist(strsplit(g, split="/")))
subtypes <- lapply(ident2, function(x) levels(factor(unlist(x))))
subtypes <- lapply(subtypes, function(s) paste(paste(s, collapse="/"), "_multiple", sep=""))
names(ident) <- subtypes
ident <- melt(do.call(rbind, ident))
ident <- unique(ident[order(ident[,1]), c(3,1)])

# Replace any sequence name in taxonomy assignment that is a member of a group of identical sequences with the name of the group
collapse.idents <- function(df) {
  within(df, {
    for (i in 1:nrow(ident)) {
      hit <- gsub(ident[i,1], ident[i,2], hit)
    }
  })
}

tax97 <- collapse.idents(tax97)
tax97bs <- collapse.idents(tax97bs)
tax100 <- collapse.idents(tax100)

tax97 <- with(tax97, tax97[order(otu, -sim), ])
tax97 <- tax97[!duplicated(tax97$otu), ]
rownames(tax97) <- tax97$otu

tax97bs <- with(tax97bs, tax97bs[order(otu, -sim), ])
tax97bs <- tax97bs[!duplicated(tax97bs$otu), ]
rownames(tax97bs) <- tax97bs$otu

tax100 <- with(tax100, tax100[order(otu, -sim), ])
tax100 <- tax100[!duplicated(tax100$otu), ]
rownames(tax100) <- tax100$otu
```

```{r build_phyloseq, include=F}
# Import OTU tables
otu97 <- otu_table(read.table("data/otus_97/97_otus.tsv", header=T, check.names=F, row.names=1,
                              skip=1, sep="\t", comment.char=""), taxa_are_rows=T)
otu97bs <- otu_table(read.table("data/otus_97_bysample/97_otus_bysample.tsv", header=T, check.names=F, row.names=1,
                              skip=1, sep="\t", comment.char=""), taxa_are_rows=T)
otu100 <- otu_table(read.table("data/otus_100/100_otus.tsv", header=T, check.names=F, row.names=1,
                               skip=1, sep="\t", comment.char=""), taxa_are_rows=T)

# Import sample data
sam <- sample_data(read.delim("data/mapping_file.txt", sep="\t", header=T, row.names=1))

# Build phyloseq objects
phy97 <- phyloseq(otu97, tax_table(as.matrix(tax97)), sam)
phy97bs <- phyloseq(otu97bs, tax_table(as.matrix(tax97bs)), sam)
phy100 <- phyloseq(otu100, tax_table(as.matrix(tax100)), sam)
```

```{r filter_dataset, include=F}
# FILTER OUT OTUS THAT ARE NOT SYMBIODINIUM
# If the top hit in NCBI does not contain the string "Symbiodinium", then this sequence is assumed to not be Symbiodinium.
poormatch97 <- readLines("data/otus_97/poormatch_IDs.txt")
poormatch97 <- data.frame(otu=str_extract(poormatch97, "denovo[^ ]*"),
                          symbio=str_detect(poormatch97, "Symbiodinium"),   # TRUE if Symbiodinium
                          stringsAsFactors = FALSE)
notsymbio97 <- poormatch97[which(poormatch97$symbio==F), 1]

poormatch97bs <- readLines("data/otus_97_bysample/poormatch_IDs.txt")
poormatch97bs <- data.frame(otu=str_extract(poormatch97bs, "denovo[^ ]*"),
                            symbio=str_detect(poormatch97bs, "Symbiodinium"),   # TRUE if Symbiodinium
                            stringsAsFactors = FALSE)
notsymbio97bs <- poormatch97bs[which(poormatch97bs$symbio==F), 1]

poormatch100 <- readLines("data/otus_100/poormatch_IDs.txt")
poormatch100 <- data.frame(otu=str_extract(poormatch100, "denovo[^ ]*"),
                           symbio=str_detect(poormatch100, "Symbiodinium"),   # TRUE if Symbiodinium
                           stringsAsFactors = FALSE)
notsymbio100 <- poormatch100[which(poormatch100$symbio==F), 1]


# Remove OTUs that are not Symbiodinium
phy97.f <- prune_taxa(!taxa_names(phy97) %in% notsymbio97, phy97)
phy97bs.f <- prune_taxa(!taxa_names(phy97bs) %in% notsymbio97bs, phy97bs)
phy100.f <- prune_taxa(!taxa_names(phy100) %in% notsymbio100, phy100)


# Filter OTUs by minimum count
# Set threshold count
n <- 10
# Identify OTUs below threshold count
taxa97 <- taxa_sums(phy97.f)[which(taxa_sums(phy97.f) >= n)]
taxa97bs <- taxa_sums(phy97bs.f)[which(taxa_sums(phy97bs.f) >= n)]
taxa100 <- taxa_sums(phy100.f)[which(taxa_sums(phy100.f) >= n)]
# Remove taxa below threshold count
phy97.f <- prune_taxa(names(taxa97), phy97.f)
phy97bs.f <- prune_taxa(names(taxa97bs), phy97bs.f)
phy100.f <- prune_taxa(names(taxa100), phy100.f)

# Filter samples by minimum count
# Set threshold number of reads
sn <- 200
# Remove samples with fewer reads than threshold
phy97.f <- prune_samples(sample_sums(phy97.f)>=sn, phy97.f)
phy97bs.f <- prune_samples(sample_sums(phy97bs.f)>=sn, phy97bs.f)
phy100.f <- prune_samples(sample_sums(phy100.f)>=sn, phy100.f)

# Filter OTUs by minimum count again in case any dropped below threshold after filtering samples
# Identify OTUs below threshold count
taxa97 <- taxa_sums(phy97.f)[which(taxa_sums(phy97.f) >= n)]
taxa97bs <- taxa_sums(phy97bs.f)[which(taxa_sums(phy97bs.f) >= n)]
taxa100 <- taxa_sums(phy100.f)[which(taxa_sums(phy100.f) >= n)]
# Remove taxa below threshold count
phy97.f <- prune_taxa(names(taxa97), phy97.f)
phy97bs.f <- prune_taxa(names(taxa97bs), phy97bs.f)
phy100.f <- prune_taxa(names(taxa100), phy100.f)

# Label clades and subtypes for filtered phyloseq object tax_tables
get.st <- function(df) {
  within(df, {
    Clade <- substr(hit, 1, 1)
    Subtype <- gsub(hit, pattern="_[A-Z]{2}[0-9]{6}", replacement="")
    Subtype <- gsub(Subtype, pattern="_multiple", replacement="")
    Subtype2 <- ifelse(as.numeric(sim)==100, paste0("'", Subtype, "'"),
                       paste0("'", rep(rle(sort(Subtype))$values, times=rle(sort(Subtype))$lengths), "'^", 
                              unlist(lapply(rle(sort(Subtype))$lengths, seq_len)))[order(order(Subtype))])
    #Subtype <- ifelse(as.numeric(sim)==100, Subtype, paste("*", Subtype, sep=""))
  })
}

tax_table(phy97.f) <- as.matrix(get.st(data.frame(tax_table(phy97.f), stringsAsFactors=FALSE)))
tax_table(phy97bs.f) <- as.matrix(get.st(data.frame(tax_table(phy97bs.f), stringsAsFactors=FALSE)))
tax_table(phy100.f) <- as.matrix(get.st(data.frame(tax_table(phy100.f), stringsAsFactors=FALSE)))
```

```{r transform_dataset, include=F}
# TRANSFORM COUNT DATA
# Convert to proportion (relative abundance)
phy97.f.p <- transform_sample_counts(phy97.f, function(x) x/sum(x))
phy97bs.f.p <- transform_sample_counts(phy97bs.f, function(x) x/sum(x))
phy100.f.p <- transform_sample_counts(phy100.f, function(x) x/sum(x))
# Apply transformation function
transform <- function(x) sqrt(x/sum(x))  # Set transformation function
phy97.f.t <- transform_sample_counts(phy97.f, transform)  # Transform data
phy97bs.f.t <- transform_sample_counts(phy97bs.f, transform) 
phy100.f.t <- transform_sample_counts(phy100.f, transform)
```

### Abstract

  Symbiotic microalgae (*Symbiodinium* spp.) strongly influence the performance, fitness, and stress-tolerance of their coral hosts, making the description of *Symbiodinium* communities in corals (and metacommunities on reefs) a routine and essential goal in coral reef research. High-throughput sequencing of ITS2 nrDNA can describe these communities with unprecedented scale, yet high intragenomic variability at this locus complicates the resolution of biologically meaningful diversity. Using samples of *Symbiodinium* from ten host taxa on shallow reefs on the north and south shores of St. John, US Virgin Islands, we demonstrate that generating operational taxonomic units by clustering *Symbiodinium* ITS2 sequences at 97% similarity within, but not across, samples collapses sequence diversity that is more likely to be intragenomic, while preserving diversity that is more likely interspecific. We utilized this ‘within-sample clustering’ approach to create metacommunity networks of host-symbiont associations in St. John, which revealed *Symbiodinium* lineages occupying 'dominant' and 'background' niches, and coral hosts that are 'flexible' or 'specific' in their associations with *Symbiodinium*, although no differences in symbiont communities were observed between north and south shore environments. These methods shed new light on important questions in coral symbiosis ecology, and demonstrate how application-specific bioinformatic pipelines can improve the analysis of metabarcoding data in microbial metacommunity studies.
  

### Introduction

  The composition of symbiotic algal communities (*Symbiodinium* spp.) in reef-building scleractinian and milleporine corals plays a major role in their biology and ecology, as the identity and functional performance of the symbionts influences emergent properties of the holobiont, including its photobiology, energetics, growth rates, and susceptibility to stress [@Rowan:2004p3383; @Cantin:2009p4376; @Jones:2011dt]. Even slight differences in the relative abundance of different symbionts may have important functional consequences for the host [@Loram:2007p3688; @Cunning:2015ja]. Moreover, variation in these communities among individuals within a host species, and within individuals over time, is an important trait modulating sensitivity of corals to environmental stress [@Putnam:2012wv] and their ability to respond in beneficial ways to changing environmental conditions by 'shuffling' or 'switching' symbionts [@Baker:2003p200]. Therefore, understanding the ecological performance of corals will greatly benefit from accurate and comprehensive descriptions of their *Symbiodinium* communities. 
  
  *Symbiodinium* identity is primarily described using genetic sequence data, including both chloroplast (cp23S and psbA^ncr^) and nuclear markers (18S, 28S, ITS1, and ITS2) [@LaJeunesse:2001p64; @Santos:2003kt; @Pochon:2012we]. Together, these markers have been used to identify nine major 'clades' within the genus *Symbiodinium* (clades A through I [@Pochon:2010p8784]), which have been further divided into 'types' based on the marker used to identify them. *Symbiodinium* species are beginning to be described based on combinations of markers, including microsatellites to establish reproductive isolation (i.e., to satisfy the biological species concept) [@LaJeunesse:2014io]. However, ecological surveys of *Symbiodinium* diversity must still rely on commonly-used marker genes, such as ITS2. Consequently, high-throughput sequencing of ITS2 increasingly is being utilized to characterize *Symbiodinium* communities with unprecedented resolution [e.g., @Arif:2014gk; @Quigley:2014dv; @Thomas:2014da; @Edmunds:2014jc; @Cunning:2015hb; @Ziegler:2017cs]. However, such analyses may create datasets consisting of millions of sequence reads and hundreds of thousands of distinct sequence variants [@Ziegler:2017cs], which places great importance on the ways in which ITS2 sequence diversity is analyzed and interpreted in relation to biological diversity.
  
  While ITS2 initially was investigated as a potential species-level marker in *Symbiodinium* [@LaJeunesse:2001p64], it is now understood that this marker is not sufficiently variable to distinguish all species within the genus [@Finney:2010p8324; @Thornhill:2013dx]. For example, the 'B1' ITS2 sequence variant is shared by *S. minutum*, *S. pseudominutum*, and *S. antillogorgium*, and potentially other species [@Parkinson:2015gt]. Moreover, the position of ITS2 within the tandemly-repeating ribosomal DNA array creates multiple ITS2 sequence variants within a single genome [@Thornhill:2007p145] that evolve through concerted evolution (e.g. [@Dover:1986ju]). In fact, concerted evolution may mask species divergence within *Symbiodinium* by maintaining ancestral sequence variants as numerical dominants in multiple derived lineages [@Thornhill:2013dx]. Together, these features of ITS2 complicate the interpretation of intragenomic versus interspecific variation, and preclude its use as a true species-level marker for *Symbiodinium*. Nevertheless, numerically dominant intragenomic variants of ITS2 are still phylogenetically informative across the genus [@LaJeunesse:2001p64], and resolve diversity at a functionally and ecologically important level. Moreover, due to the large quantity of existing sequence data for comparative analysis (e.g., [@Franklin:2012jn; @Tonk:2013ey]), and the relative ease of amplifying and sequencing ITS2, it remains an essential and powerful marker for the *Symbiodinium* system. Therefore, it is important to develop best practices in the bioinformatic analysis and interpretation of ITS2 metabarcoding surveys of *Symbiodinium*.
  
  Here, we describe a *Symbiodinium* metacommunity associated with scleractinians and a *Millepora* hydrocoral in St. John, U.S. Virgin Islands, with the objectives of: 1) developing an appropriate bioinformatic approach for *Symbiodinium* ITS2 metabarcoding surveys, and 2) exploring network analysis and statistical applications for such datasets that may advance understanding of *Symbiodinium* metacommunity ecology. In addressing the first objective, we test the ability of 'within-sample clustering' (i.e. clustering sequences at 97% similarity within each sample independently) to generate biologically relevant operational taxonomic units (OTUs) from ITS2 metabarcoding data. Specifically, this approach addresses the fact that dominant ITS2 sequences from different *Symbiodinium* species may be more similar to one another than intragenomic variants within one *Symbiodinium* [@Thornhill:2013dx; @Arif:2014gk; @Parkinson:2015gt]. Therefore, clustering all sequences together underestimates diversity by collapsing different species into the same OTU, while treating each ITS2 sequence as a unique *Symbiodinium* type overestimates diversity due to intragenomic variation. Within-sample clustering may better approximate true diversity by exploiting key assumptions regarding the distribution of symbionts among samples, and sequences among symbionts. These assumptions include: 1) that most coral colonies are dominated by a single *Symbiodinium* type [@LaJeunesse:2011us; @Pettay:2011wt; @Baums:2014hj], and 2) that different numerically dominant ITS2 sequence variants diagnose different *Symbiodinium* types, even when they differ by only 1 nucleotide [@Sampayo:2009p3422]. These assumptions suggest that when closely related ITS2 sequences (i.e., that are >97% similar; @Arif:2014gk) occur within the same sample, they are more likely to be intragenomic variants, but when they are numerically dominant in different samples, they are more likely to represent distinct *Symbiodinium* taxa. Accordingly, clustering sequences at 97% similarity within each sample independently may collapse variability that is more likely to be intragenomic (i.e., occurring within a sample), while maintaining variability that is more likely to be interspecific (i.e., occurring in different samples). 
  
  Using within-sample clustering, we analyzed *Symbiodinium* communities in ten host species across the north and south shores of St. John. We explore ecological patterns that can be revealed by large-scale metabarcoding datasets, including 1) testing for whole-community differences associated with north and south shore locations, 2) analyzing coral-*Symbiodinium* metacommunity association networks, and 3) quantifying the variability in symbiont communities (i.e., beta diversity, or 'symbiosis flexibility') among individuals within a host species. The large quantity of data, and the kinds of analytical approaches facilitated by ITS2 metabarcoding, have the potential to revolutionize our understanding of *Symbiodinium* metacommunity ecology in reef corals and other taxa harboring similar symbionts. More generally, the present study demonstrates how ecological knowledge can inform bioinformatic analyses using marker genes for which sequence diversity does not map directly to species.
  

### Materials and Methods

#### Sample collection and environmental conditions

  Eighty-four tissue samples were collected from scleractinian and milleporine corals sampled at seven sites around St. John, US Virgin Islands, USA (Fig. 1), between August 7th and 9th, 2012. Samples were collected on snorkel from 3-7m depth, and all colonies to be sampled were selected haphazardly in upward-facing locations exposed to ambient irradiance. Tissue biopsies (~5-mm diameter and ~5-mm depth) were removed from coral colonies with sterile metal borers, and the small holes were plugged with non-toxic modeling clay. 

  Broad-scale differences in the hydrographic regimes at points representative of the north and south shores of St. John (North: 18.368141°N, 64.711967°W; South: 18.315465°N, 64.726840°W) were evaluated from remote sensing tools to determine sea surface temperature (°C), surface chlorophyll concentrations (mg chl a m^-3^), and integrated wave exposure on a power scale (J m^-3^). These features were selected as they corresponded to anecdotal observations of differences between the water masses on both shores, and conditions known to affect coral performance and *Symbiodinium* biology (i.e., SST [@Coles:1977p7350] and wave regime [@Atkinson:1994un]), and mediate the supply of food resources that play important roles in coral nutrition [@Houlbreque:2009gh]. SST and chlorophyll were determined using data from the MODIS sensor on the Aqua satellite. SST was evaluated from nighttime records, and chlorophyll from ocean colour, both of which were obtained at 1 km spatial resolution for each month, averaged over the years 2003-2010, from the IMaRS website (http://imars.usf.edu/modis); if data were not available at the chosen coordinates (above), values from the nearest available pixel were used. Boxplots were generated from the 12 monthly climatological values obtained for each response (Fig. 1).

  Wind-driven wave exposure on a power scale for a given site is dependent on the wind patterns and the configuration of the coastline, which defines the fetch. To calculate wave exposure, wind speed and direction at each location were acquired from the QuickSCAT (NASA) satellite scatterometer from 1999 to 2008 at 25 km spatial resolution [@Hoffman:2005dz]. Coastline data were obtained from the Global Self-consistent, Hierarchical, High-resolution, Shoreline (GSHHS v 2.2) database, which provides global coastline at 1:250,000 scale [@Wessel:1996wk]. From these data, wave exposure was calculated using the methods based on wave theory (after Chollett et al. 2012) for 32 fetch directions (equally distributed through 360°) and coastline data at full resolution (xx km). Total wave exposure (summed over all directions) was calculated in R using the packages maptools [@maptoolsToolsfor:_dYm01UK], raster [@rasterGeographicD:524kz2M0], rgeos [@rgeosInterfaceto:8WVz_nNk], and sp [@Bivand:2013ux].
  
#### ITS2 sequencing and bioinformatics
  
  Coral tissue samples were preserved in ~500 µL Guanidinium buffer (50% w/v guanidinium isothiocyanate; 50 mM Tris pH 7.6; 10 μM EDTA; 4.2% w/v sarkosyl; 2.1% v/v β-mercaptoethanol) and shipped to the Hawaii Institute of Marine Biology (HIMB). Genomic DNA was extracted from each coral tissue sample following a guanidinium-based extraction protocol [@Cunning:2015hb], and sent to Research and Testing Laboratory (Lubbock, TX) for sequencing of ITS2 amplicons ('itsD' and 'its2rev2' primers from @Stat:2009p7366) on the Illumina MiSeq platform with 2x300 paired-end read chemistry. 

  Paired reads from each sample (provided in .fastq format by Research and Testing Laboratory) were merged using illumina-utils software [@Eren:2013hh] with an enforced Q30-check and an overlap ≥150 bp with ≤3 mismatches required to retain a sequence. Chimeric sequences were removed using usearch61 [@Edgar:2010cv] implemented in QIIME [@Caporaso:2010jf]. Primers were trimmed using cutadapt [@Martin:2011eu] allowing 3 mismatches, and only sequences with both forward and reverse primer matches and length ≥250 bp after trimming were retained. Subsequently, three different clustering approaches, each based on the uclust algorithm [@Edgar:2010cv] and implemented in QIIME, were used to group ITS2 sequences into operational taxonomic units (OTUs): 1) clustering at 100% identity, 2) clustering at 97% identity across samples (i.e., sequences from all samples clustered together), and 3) clustering at 97% identity within samples (i.e., sequences from each sample clustered independently). For each 97% cluster, the most abundant sequence variant was chosen as the representative sequence, and within-sample clusters were merged across samples if their representative sequences were 100% identical. After removing singleton clusters, representative sequences for each OTU were assigned taxonomy by searching a custom reference database of *Symbiodinium* ITS2 sequences using the Needleman-Wunsch global alignment algorithm implemented in the Biostrings package [@BiostringsStringo:Fbauxymn] in R [@RALanguageandEn:2014wf]. Each OTU was then assigned a name corresponding to the reference sequence(s) with the highest alignment score; if the match was <100%, the assignment was given a unique superscript. If the match was <90%, the sequence was blasted to NCBI and omitted if the top hit did not contain "*Symbiodinium*".
  
  The reference database comprised `r system("grep -c '>' data/ITS2db_trimmed_derep.fasta")` *Symbiodinium* ITS2 sequences downloaded directly from NCBI. These sequences included those used in previous reference databases [@Cunning:2015hb] supplemented with additional sequences of *Symbiodinium* from Caribbean corals [@Finney:2010p8324; @Green:2014ev]. Reference sequences were separated and aligned by clade using muscle [@Edgar:2004bo], and trimmed to equal length using the o-smart-trim command from oligotyping software [@Eren:2013fp], before being reconcatenated into a single fasta file. The bioinformatic pipeline used here was implemented using a series of bash scripts and a Makefile, which can be found in the data archive accompanying this paper and executed to fully reproduce the present analysis.

#### *Symbiodinium* data analysis

  OTU tables, sample metadata, and taxonomic data were imported into R using the phyloseq package [@McMurdie:2013ky] to facilitate downstream analyses. OTU tables were filtered to remove any OTU that was not observed at least `r n` times, or any sample with < `r sn` sequences, and counts were transformed to relative abundance. Permutational analysis of variance (PERMANOVA) was used to test for differences in *Symbiodinium* community composition between the north and south shores of St. John within each coral species.
  
  Network analysis and visualization was performed in R using the igraph package [@Csardi:2006vh]. Networks were created featuring 'dominant' (>50% relative abundance) and 'abundant' (>1% relative abundance) OTUs, with weighted edges proportional to the number of individuals within a species in which a symbiont OTU occurred. A 'background' symbionts network was also created with unweighted edges defined based on the presence of a symbiont at <1% relative abundance in at least one individual within the species. To simplify visualization of the background symbiont network, clade D OTUs were merged into a single node under the assumption that clade D in the Caribbean comprises a single species (*Symbiodinium trenchii*; [@LaJeunesse:2014io]), and symbiont nodes connected to ≤ 2 coral species were removed from the network. All network layouts were constructed based on the Fruchterman-Reingold algorithm [@Fruchterman:1991ko].
  
  Beta diversity (sensu [@Anderson:2006fs]) was calculated as the multivariate dispersion of samples within a coral species. Principal coordinate analysis of Bray-Curtis dissimilarities of square-root transformed OTU counts was used to calculate average distance-to-centroid values for each species, which were then compared statistically by a permutation test. This analysis was implemented using the betadisper function in the vegan package [@veganCommunityEco:uw], based on [@Anderson:2006fs].


### Results

#### Sample collection and environmental data

  A total of 84 coral samples representing ten different host species were collected across two sites on the north shore and five sites on the south shore of St. John (Fig. 1). The sequence data generated from these samples comprised 1,490,813 sequences after merging paired reads, removing chimeric sequences, and trimming primers. After clustering, OTUs with <`r n` sequences and samples with <`r sn` sequences were excluded, leaving `r nsamples(phy97.f)` samples for downstream analysis.
  
  The environmental conditions broadly characterizing the north and south shores are presented in Fig. 1. From 2003 to 2010, the north shore of St. John was characterized by slightly lower sea surface temperatures, higher chlorophyll a concentrations, and lower wave exposure, relative to the south shore.

```{r Fig1, fig.height=3, fig.width=6.85, fig.cap="Figure 1. Sampling of *Symbiodinium* communities from ten different host species (embedded histogram) on the north and south shores of St. John, US Virgin Islands, and the broad-scale physical conditions associated with these two environments. Mapped points along each shore indicate locations from which biological samples were collected, while the 'N' and 'S' points indicate locations from which environmental data representative of the north and south shores were obtained. Boxplots show the sea-surface temperatures (top), chlorophyll A concentrations (middle), and wave exposure (bottom) recorded at these two locations."}
library(rgdal)
stjmap <- readOGR("data/coastline/zipfolder", layer="Coastal_Coastline_line", verbose=F)  # units are in meters
stjmap <- spTransform(stjmap, CRS("+proj=longlat +datum=WGS84"))  # transform to lat/long

sitecoords <- data.frame(
  name=c("Booby Rock", "Tektite", "Kiddel", "Groot Pan", "Yawzi Point", "Anna's Point", "Haulover Bay"),
  lat=c(18.30327, 18.31057, 18.30770, 18.31067, 18.31515, 18.36710, 18.35008),
  lon=c(-64.71057, -64.72203, -64.71398, -64.71813, -64.72513, -64.73337, -64.67933)
)

viers <- c(-64.722697, 18.322277)

#pdf(file="figures/map.pdf", height=3, width=6.85)
layout(mat=matrix(c(1,1,6,2,1,1,5,3,1,1,5,4), nrow=3, byrow=T))

# PLOT MAP
par(mar=c(0,0,0,0), oma=c(0,0,0,0), cex=1, xpd=NA)
plot(stjmap, lwd=1.5, col="gray40",
     xlim=c(-64.77, -64.70), ylim=c(18.303, 18.370))
#text(-64.735, 18.34, expression(italic("       St. John\nU.S. Virgin Islands")), col="gray40")
axis(side=3, at=seq(-64.78, -64.66, 0.01), line=0, tck=0.02, labels=FALSE)
axis(side=3, at=seq(-64.78, -64.66, 0.04), line=-2, lwd=0, cex.axis=0.6)
axis(side=2, at=seq(18.3, 18.4, 0.01), line=0, tck=0.02, labels=FALSE)
axis(side=2, at=seq(18.3, 18.4, 0.02), line=-2, lwd=0, cex.axis=0.6)
with(sitecoords, {
  points(lon, lat, pch=21, bg="black", cex=0.8)
  text(lon, lat, name, pos=c(4,2,4,4,4,1,3), cex=0.8)
})
NS <- c(-64.711967, 18.368141)
SS <- c(-64.726840, 18.315465)
points(NS[1], NS[2], pch="N", cex=0.9, bg="gray80", col="black", lwd=1)
points(SS[1], SS[2], pch="S", cex=0.9, bg="gray40", col="black", lwd=1)
points(NS[1], NS[2], pch=1, cex=2.2, bg="gray80", col="black", lwd=1)
points(SS[1], SS[2], pch=1, cex=2.2, bg="gray40", col="black", lwd=1)

# Plot histogram of sample collections
par(new=T)
samdat <- data.frame(sample_data(phy97))
nsam <- with(samdat, aggregate(Species, by=list(Species=Species, Shore=Shore), FUN=length))
nsam <- dcast(nsam, Species~Shore)
rownames(nsam) <- nsam$Species
rownames(nsam) <- str_replace_all(rownames(nsam), 
                c("alcicornis"="M.alc.", "annularis"="O.ann.", "astreoides"="P.ast.","cavernosa"="M.cav.",
                  "cylindrus"="D.cyl.", "fragum"="F.fra.", "furcata"="P.fur.", "radians"="S.rad.",
                  "siderea"="S.sid.", "strigosa"="P.str."))
nsam <- nsam[,2:3]
par(mar=c(7,5,5,6), mgp=c(0.5,0.2,0))
barplot(t(as.matrix(nsam)), las=2, ylab="# samples", xpd=NA, cex.axis=0.5, cex.names=0.5, cex.lab=0.5, tcl=0.1)
legend("topleft", pch=22, pt.bg=c("gray90", "gray40"), pt.cex=1, cex=0.5, legend=c("N","S"), bty="n", inset=c(0.12,-0.1), xpd=T, y.intersp=0.8, x.intersp=0.75)

# Plot environmental data
envdata <- read.csv("data/STJ_envdata2.csv")
envdata$Site <- factor(envdata$Site, levels=c("S", "N"))

#SST
par(cex=0.7, mar=c(3,0,1,2), mgp=c(1,0.25,0), xpd=NA, cex.axis=0.7, cex.lab=0.7, tcl=-0.25)
plot(NA, xlim=c(25,30), ylim=c(0.5,2.5), bty="n", yaxt="n", ylab="", xlab="SST (°C)", xaxt="n")
axis(side=1, at=seq(25,30,1))
text(par("usr")[1], c(1,2), c("S","N"), pos=2, offset=1)
points(rep(par("usr")[1]-strwidth("c.m"),2), c(1,2), pch=1, cex=2.5)
boxplot(envdata$SST ~ envdata$Site, range=0, add=T, axes=F, horizontal=T)

#chlA
plot(NA, xlim=c(0,0.1), ylim=c(0.5,2.5), bty="n", yaxt="n", ylab="", xlab="ChlA (mg m-3)", xaxt="n")
axis(side=1, at=seq(0,0.1,0.01))
text(par("usr")[1], c(1,2), c("S","N"), pos=2, offset=1)
points(rep(par("usr")[1]-strwidth("c.m"),2), c(1,2), pch=1, cex=2.5)
boxplot(envdata$chlA ~ envdata$Site, range=0, add=T, axes=F, horizontal=T)

#Wave exposure
we <- list(N=3.234787362,	S=4.025022708)
barplot(c(we$N, we$S), horiz=T, xlim=c(0,5), ylab="", xlab="Wave exposure (ln J m-3)")
text(par("usr")[1], c(1,2), c("S","N"), pos=2, offset=1, xpd=NA)
points(rep(par("usr")[1]-strwidth("c.m"),2), c(1,2), pch=1, cex=2.5, xpd=NA)

#dev.off()
```

#### Comparison of clustering approaches

   The number of OTUs resolved, as well as the number of sequences per OTU and per sample, depended on the clustering approach used (Table 1 and Fig. S1). More OTUs were resolved as the clustering resolution increased (i.e., 97% OTUs across samples < 97% OTUs within samples < 100% OTUs), with fewer reads per OTU (Table 1). The number of sequences per sample was less for the 100% OTU approach because more OTUs were filtered out of the dataset by the minimum threshold count of `r n` reads per OTU.

```{r Table1, cache=F}
# Compute summary statistics 
stats97.f <- data.frame(`97% OTUs<br>(across samples)`=t(phystats(phy97.f)), check.names=F)
stats97bs.f <- data.frame(`97% OTUs<br>(within samples)`=t(phystats(phy97bs.f)), check.names=F)
stats100.f <- data.frame(`100% OTUs`=t(phystats(phy100.f)), check.names=F)

knitr::kable(cbind(stats97.f, stats97bs.f, stats100.f)[c(2, 3, 6, 8), ], align="c",
             caption = "Table 1. Summary statistics for each clustering approach.")
```

  A subset of samples was selected for comparative analysis of the community structure resolved by the three clustering approaches (Fig. 1). 97% across-sample clustering assigned the same dominant OTU to samples dominated by different sequence variants, while 97% within-sample clustering assigned distinct dominant OTUs to these samples. Because the outcomes of the latter approach are more consistent with the current understanding of ITS2 sequence diversity as it links to *Symbiodinium* biology and ecology (see Discussion), the remainder of the results is presented using the 97% within-sample clustering approach. Additional detailed comparisons of the outcomes of each clustering approach for each coral species are presented in the Supplementary Information.
    
```{r Fig2, fig.width=6, fig.height=4, fig.align="center", fig.cap="Figure 2. Comparison of dominant OTUs assigned by 100%, 97% across-sample, and 97% within-sample clustering approaches. A.) The composition of unique sequence variants (i.e., 100% OTUs) in each sample, sorted by relative abundance, and filled by unique colors corresponding to unique sequence variants. The dominant OTUs within each sample assigned by B.) 97% across-sample clustering and C.) 97% within-sample clustering are shown by rectangles below each bar, with fill colors matching the unique sequence variants presented in (A.) to indicate the representative sequence of the assigned OTU. Colored rectangles that span multiple bars indicate that the corresponding samples were assigned the same dominant OTU. Rectangles are annotated with the taxonomic assignment of the OTU (see Methods)."}
set.seed(42432) 
par(mfrow=c(1,1))
otubarplot2(samples=c("64","72","29","45","73","59","47","26","48","66","63"))
```


#### *Symbiodinium* community composition

```{r cladeAbund, include=F}
# Calculate relative abundance of each clade
cladeAbund <- aggregate(data.frame(RelAbund=rowSums(otu_table(phy97bs.f.p))),
                        by=list(Clade=data.frame(tax_table(phy97bs.f.p))$Clade), FUN=sum)
cladeAbund$Prop <- round(prop.table(cladeAbund$RelAbund) * 100, 1)
cladeAbund$Notus <- table(data.frame(tax_table(phy97bs.f.p))$Clade)
clademax <- aggregate(data.frame(max=apply(otu_table(phy97bs.f.p), 1, FUN=max)),
                      by=list(Clade=data.frame(tax_table(phy97bs.f.p))$Clade), FUN=max)
```

  
  Within the set of coral samples obtained in this study, clade B had the highest relative abundance (`r with(cladeAbund, Prop[Clade=="B"])`%), followed by clade C (`r with(cladeAbund, Prop[Clade=="C"])`%), A (`r with(cladeAbund, Prop[Clade=="A"])`%), D (`r with(cladeAbund, Prop[Clade=="D"])`%), and G (`r with(cladeAbund, Prop[Clade=="G"])`%. The number of OTUs within each clade followed a similar pattern, with the highest number in clade B (`r with(cladeAbund, Notus[Clade=="B"])`), followed by clade C (`r with(cladeAbund, Notus[Clade=="C"])`), A (`r with(cladeAbund, Notus[Clade=="A"])`), D (`r with(cladeAbund, Notus[Clade=="D"])`), and G (`r with(cladeAbund, Notus[Clade=="G"])`). The distribution of these clades within each sample is shown in Fig. 3.
  
```{r Fig3, fig.height=6, fig.width=4, fig.align="center", fig.cap="Figure 3. *Symbiodinium* community composition of each sample. Samples are plotted as horizontal bars, sorted by species and shore of collection (north vs. south). Individual segments of each bar represent 97% within-sample OTUs, colored by *Symbiodinium* clade identity."}
#png(file="crap.png", width = 4, height=6, units="in", res=300)
par(mfrow=c(1,1), mar=c(2, 1.5, 2, 5), lwd=0.1, cex=1)
# Plot composition of 97% within-sample OTUs colored by clade
composition(phy97bs.f.p, col=taxcolors[factor(data.frame(tax_table(phy97bs.f.p))[order(data.frame(tax_table(phy97bs.f.p))$Subtype), ]$Clade, levels=c("A","B","C","D","F","G"))], legend=T)
#dev.off()
```

#### Differences in *Symbiodinium* between shores

  No significant differences in *Symbiodinium* community composition between the north and south shores were detected for any host species (Table 2). However, some qualitative differences were apparent: *Porites furcata* colonies were dominated by either clade A or clade C *Symbiodinium* on the south shore, but clade C only on the north shore (difference between shores PERMANOVA: p=0.056). *Siderastrea siderea* was dominated by either clade C or clade D on the south shore, but clade C only on the north shore (difference between shores PERMANOVA: p=0.071). 

```{r Table2, cache=T } 
set.seed(43789)  
shorestats97bs <- perms(phy97bs.f.p, group="Species", trt="Shore")
shorestats97bs$Species <- str_replace_all(
  shorestats97bs$Species,
  c("alcicornis"="*Millepora alcicornis*",
    "annularis"="*Orbicella annularis*",
    "astreoides"="*Porites astreoides*",
    "cavernosa"="*Montastraea cavernosa*",
    "cylindrus"="*Dendrogyra cylindrus*",
    "fragum"="*Favia fragum*",
    "furcata"="*Porites furcata*",
    "radians"="*Siderastrea radians*",
    "siderea"="*Siderastrea siderea*",
    "strigosa"="*Pseudodiplora strigosa*")
)
knitr::kable(shorestats97bs, digits=3, row.names=F, 
             caption="Table 2. Bray-Curtis dissimilarities within and between shores for *Symbiodinium* communities in each coral species and PERMANOVA tests for a difference between shores.")



#df <- data.frame(otu_table(phy97bs.f.p), check.names=F)

#clade.otutab <- aggregate(df, by=list(Clade=data.frame(tax_table(phy97bs.f.p))$Clade), FUN=sum)
#rownames(clade.otutab) <- clade.otutab[,1]
#clade.otutab <- clade.otutab[,-1]

#clade.phy <- phyloseq(otu_table(clade.otutab, taxa_are_rows=T), sample_data(sam))
#perms(clade.phy, group="Species", trt="Shore")

```

#### Network analyses of *Symbiodinium* metacommunity
```{r networks,  cache=F, include=FALSE}
# Create dominant symbionts network
set.seed(12374)
domnet <- sppnet(phy=phy97bs.f.p, plot=F,
                 fun=function(x) length(x[x>0.5])/length(x), 
                 layout=layout.fruchterman.reingold)
# Calculate number of dominant symbionts connected to each species
dom.Nedges <- sapply(levels(data.frame(sample_data(phy97bs.f.p))$Species),
                     FUN=function(x) length(E(domnet)[from(x)]))
dom.Nedges <- sort(dom.Nedges, decreasing=T)

# Create abundant symbionts network
set.seed(12374)
abunet <- sppnet(phy=phy97bs.f.p, plot=F,
                 fun=function(x) length(x[x>0.01])/length(x),
                 layout=layout.fruchterman.reingold)
# Calculate number of abundant symbionts connected to each species
abu.Nedges <- sapply(levels(data.frame(sample_data(phy97bs.f.p))$Species),
                     FUN=function(x) length(E(abunet)[from(x)]))
abu.Nedges <- sort(abu.Nedges, decreasing=T)

# Create background symbionts network
# filter out OTUs that only occurred in one sample
phy97bs.f.p2 <- filter_taxa(phy97bs.f.p, function(x) sum(x>0) > 1, prune=TRUE)
bkgnet <- sppnet(phy=phy97bs.f.p2, plot=F,
                 fun=function(x) any(x>0 & x<0.01),
                 layout=layout.fruchterman.reingold)
# Merge D nodes (assuming they are all D1a/trenchii)
# Get original node map (sequence starting at one to the total number of nodes)
nodemap <- seq(1:length(V(bkgnet)))
# Make a copy of the nodemap that will be modified
nodemapnew <- nodemap
# Find all nodes beginning with D, and give them the same node ID number
nodemapnew[grep("^D", V(bkgnet)$Subtype)] <- min(grep("^D", V(bkgnet)$Subtype))
# Replace the now missing values in the node map so node IDs remain sequential
while (max(setdiff(nodemap, nodemapnew)) < max(nodemap)) {
  nodemapnew[which(nodemapnew==max(setdiff(nodemap, nodemapnew))+1)] <- min(setdiff(nodemap, nodemapnew))
}
# Contract the network using the new nodemap (nodes with same node ID are merged)
bkgnet2 <- contract(bkgnet, nodemapnew, vertex.attr.comb=list(size="sum", "last"))
# Simplify network so that multiple edges are combined
bkgnet2 <- simplify(bkgnet2, remove.multiple=T, edge.attr.comb=list(weight="mean", width="mean", "ignore"))
# remove symbionts that are only connected to one or two coral species
bkgnet3 <- delete.vertices(bkgnet2, degree(bkgnet2)<=2) # & V(net2)$type==2)
V(bkgnet3)$size <- ifelse(is.na(V(bkgnet3)$Clade), 15, 1.5*degree(bkgnet3)^1.5)
E(bkgnet3)$width <- E(bkgnet3)$weight * 2

# Calculate number of edges for background symbionts
nspp <- sapply(V(bkgnet3)[V(bkgnet3)$type==2], function(x) length(E(bkgnet3)[from(x)]))
names(nspp) <- V(bkgnet3)$label[match(names(nspp), V(bkgnet3)$name)]
nspp <- sort(nspp, decreasing=T)
```

 
  Patterns of association between hosts and *Symbiodinium* were analyzed by constructing three networks focusing on "abundant" (>1% relative abundance; Fig. 4), "dominant" (>50% relative abundance; Fig. 5A), and "background" (<1% relative abundance; Fig. 5B) symbionts in each host species, based on 97% within-sample clustering. (Networks for individuals within each species are presented in Supplementary Materials). In the network for abundant symbionts (Fig. 4), `r sum(V(abunet)$type==2)` OTUs were observed. *P. strigosa* associated with the greatest number of OTUs (n=`r abu.Nedges["strigosa"]`), followed by *F. fragum* (n=`r abu.Nedges["fragum"]`), *M. alcicornis* (n=`r abu.Nedges["alcicornis"]`), and *D. cylindrus* (n=`r abu.Nedges["cylindrus"]`); these primarily comprised OTUs closely related to B1 and B19. In contrast, only a single OTU occurred at >1% relative abundance in both *P. astreoides* (A4) and *M. cavernosa* (C3). All coral species hosted at least one "abundant" *Symbiodinium* OTU that was also abundant in at least one other coral species, except for *S. radians*. This species was dominated by C46, but also contained *Symbiodinium* closely related to B1 and B19, as well as B5.  
  
  In the network for dominant symbionts, `r sum(V(domnet)$type==2)` different *Symbiodinium* OTUs were included, defined as those occurring at >50% relative abundance within a host (Fig. 5A). A single OTU identical to *Symbiodinium* B1 was the most prevalent dominant symbiont, dominating all *D. cylindrus* samples and many *P. strigosa* (4 of 9), *O. annularis* (2 of 4), *M. alcicornis* (6 of 10), and *F. fragum* (8 of 9) samples. Other closely-related clade B OTUs (similar to B1, B1d, and B19) occasionally dominated *P. strigosa*, *M. alcicornis*, and *F. fragum*, while *O. annularis* was dominated just as often by *Symbiodinium* C7. After B1, the next most prevalent dominant symbionts were *Symbiodinium* C3, which dominated all *M. cavernosa* and most *S. siderea* (7 of 10), and *Symbiodinium* A4, which dominated all *P. astreoides* and 3 of 10 *P. furcata*. *P. furcata* and *S. siderea* were each occasionally dominated by other symbiont OTUs, including C45 or C45a (*P. furcata*), and C1 or D1 (*S. siderea*). All coral species were dominated by at least one *Symbiodinium* OTU that also dominated at least one other coral species, except for *S. radians*, which was exclusively dominated by C46. 
  
  There were `r sum(V(bkgnet3)$type==2)` symbionts that occupied a 'background' niche, defined as <1% relative abundance in three or more coral species (Fig. 5B). A member of *Symbiodinium* clade D (which has been formally described as *S. trenchii* [@LaJeunesse:2014io]) was found at background levels in the greatest number of host species examined (`r nspp["D1"]`), followed by A4 and C3 (both were in `r nspp["C3"]` taxa). C31, C46, and B1 were each detected at background levels in 4 host species.

```{r Fig4, fig.width=5, fig.height=5, fig.align="center", fig.retina=1, out.width=5, fig.cap="<br>Figure 4. Network analysis of abundant *Symbiodinium* taxa in each coral species. *Symbiodinium* OTUs (circular nodes) are connected to each coral species (square nodes) in which they ever occurred at >1% relative abundance within an individual; thickness of edges (i.e., links between coral and symbiont nodes) is relative to the proportion of individuals within the coral species in which the *Symbiodinium* OTU was present at >1%. Symbiont nodes are colored by clade identity, and sized proportionally to the number of coral species they dominated."}

# Plot abundant symbionts network
par(mfrow=c(1,1), mar=c(0,0,0,0), lwd=1)
V(abunet)$size <- ifelse(is.na(V(abunet)$Clade), 10, 13*(degree(abunet))^0.4)
set.seed(12374)
l <- layout.fruchterman.reingold(abunet)
l <- norm_coords(l, ymin=-1, ymax=1, xmin=-1, xmax=1)
plot(abunet, rescale=F, layout=l, edge.curved=0.1, 
     vertex.label.cex=ifelse(is.na(V(abunet)$Clade), V(abunet)$size/15, V(abunet)$size/18),
     vertex.label.color="black") 
```

```{r Fig5, fig.width=7.20472, fig.height=3.60236, fig.align="center", fig.cap="Figure 5. Network analysis of dominant (A) and background (B) *Symbiodinium* taxa in each coral species (see Materials and Methods for details on network construction). Symbiont nodes are colored by clade identity, sized proportionally to the number of coral species to which they are connected."}

par(mfrow=c(1,2), mar=c(0.1,1,1.9,1), lwd=1, xpd=NA) 
# Plot dominant symbionts network
set.seed(12374)
l <- layout.fruchterman.reingold(domnet)
l <- norm_coords(l, ymin=-1, ymax=1, xmin=-1, xmax=1)
V(domnet)$size <- ifelse(is.na(V(domnet)$Clade), 16, 18*(degree(domnet))^0.5)
plot(domnet, rescale=F, layout=l, edge.curved=0.1, vertex.label.cex=V(domnet)$size/25,
     vertex.label.color="black")
title("A.) Dominant symbionts", adj=0)

# Plot background symbionts network
set.seed(42384)
l3 <- layout.fruchterman.reingold(bkgnet3)
l3 <- norm_coords(l3, ymin=-1, ymax=1, xmin=-1, xmax=1)
V(bkgnet3)$size <- ifelse(is.na(V(bkgnet3)$Clade), 16, 8*(degree(bkgnet3))^0.8)
plot(bkgnet3, rescale=F, layout=l3, edge.curved=0.1, vertex.label.cex=V(bkgnet3)$size/25,
     vertex.label.color="black")
points(0.9,0.9, pch=0, cex=3.5)
text(0.9,0.9, "Oa", cex=1)
title("B.) Background symbionts", adj=0)
```


#### Symbiotic flexibility of host species

  Symbiont communities from each individual host were ordinated in multivariate space, and the dispersion of individuals within each host species (i.e., average distance to centroid) was calculated as a metric of symbiosis 'flexibility', with higher values indicating hosts with greater variability in symbiont community composition among individuals, and lower values indicating hosts with greater uniformity. This metric revealed significant differences in symbiotic flexibility among host species (Fig. 6). The highest distance to centroid was found in *P. strigosa*, *P. furcata*, *M. alcicornis*, *S. siderea*, and *M. annularis*, and the lowest was found in *M. cavernosa*. Flexibility was also low in *D. cylindrus*, *S. radians*, and *P. astreoides*, and intermediate in *F. fragum*.

```{r Fig6, cache=F, fig.width=3.5, fig.height=3.5, fig.cap="Figure 6. Symbiotic flexibility in coral species quantified as multivariate dispersion of *Symbiodinium* community composition (mean distance to centroid ± SE. Host taxa that do not share a letter are significantly different (p < 0.05)."}
betad97bs <- betad(phy97bs.f.t, group="Species")

par(mar=c(7,3,1,1), mgp=c(1.8,0.1,0), tcl=0.25)
with(betad97bs$sambdsumm.ord, {
  plot(mean, type="n", ylim=c(0, 0.65), ylab="Distance to centroid", xaxt="n", xlab="")
  arrows(1:10, mean - se, 1:10, mean + se, length=0.05, angle=90, code=3)
  points(1:10, mean, cex=1, pch=21, bg="white")
  axis(side=1, at=1:10, cex.axis=0.75, labels=NA)
  text(1.4:10.4, -0.04, srt=45, xpd=T, pos=2, font=3,
       labels=str_replace_all(levels(sam$Species)[order(betad97bs$sambdsumm$mean, decreasing=T)],
                              c("alcicornis"="M. alcicornis",
                                "annularis"="O. annularis",
                                "astreoides"="P. astreoides",
                                "cavernosa"="M. cavernosa",
                                "cylindrus"="D. cylindrus",
                                "fragum"="F. fragum",
                                "furcata"="P. furcata",
                                "radians"="S. radians",
                                "siderea"="S. siderea",
                                "strigosa"="P. strigosa")))
  text(1:10, mean + se + 0.03, labels=betad97bs$saml$Letters[as.character(group)], cex=0.5)
})
```

### Discussion

#### Bioinformatic analysis of ITS2 diversity

  The observed structure of *Symbiodinium* communities in a variety of host species was impacted by the bioinformatic approach used to analyze ITS2 sequence data. Specifically, whether sequences were clustered into operation taxonomic units (OTUs) collectively across the entire dataset, or independently within each sample, affected the number and taxonomic assignment of OTUs. Clustering sequences across samples reduced the total number of OTUs by 98% relative to the number of OTUs detected with no clustering (Table 1). Within-sample clustering resolved intermediate diversity, with 13% more OTUs than across-sample clustering. Comparing taxonomic assignments of the OTUs generated by these approaches demonstrates that across-sample clustering often assigns the same dominant OTU to samples with different dominant sequence variants (Fig. 2), while within-sample clustering assigns them distinct OTUs that reflect the dominant variant in the sample. This occurs because OTU identity is determined by the most abundant member sequence, and when members occur across many samples (i.e., during across-sample clustering), each member's abundance reflects the number (and sequencing depth) of samples in the dataset that contain it. Therefore, a particular sequence variant from a more extensively-sampled and/or deeply-sequenced host may determine the identity of an OTU that includes closely-related but different sequences in other samples, even though they do not actually contain the given sequence. Such merging of OTUs across samples is undesirable, since even single nucleotide differences in a sample's dominant ITS2 variant may indicate different *Symbiodinium* species [@Sampayo:2009p3422]. 
  
  Therefore, even though clustering at 97% similarity collapses intragenomic variation within a single *Symbiodinium* [@Arif:2014gk], when such clustering is applied across a dataset potentially comprised of many closely related *Symbiodinium*, it is likely to also collapse interspecific variation and underestimate diversity. This limitation was also encountered by @Arif:2014gk when clustering closely-related clade C sequences across samples from multiple hosts: despite being comprised mostly of C41 variants, ~3,000 ITS2 sequences from *Acropora hemprichii* were subsumed into a single OTU that was named C1 after the dominant variant among the ~7,000 sequences from *Pocillopora verrucosa* with which they were clustered. In this case (and as shown in Fig. 2), clustering sequences across samples leads to the conclusion that samples with different dominant sequence variants are dominated by the same *Symbiodinium* OTU. This outcome occurs because the identity of OTUs in any given sample may be determined by other samples in the dataset; indeed, using this approach, the same sample and sequence assemblage may receive different OTU assignments if it were part of a different set of samples. 
  
  Meanwhile, clustering sequences within each sample independently assigns OTUs that reflect only the assemblage of sequences within that sample, and therefore does not depend on the presence or abundance of sequences from other samples included in the analysis. The outcome of within-sample clustering of ITS2 sequences as applied herein better reflects patterns of *Symbiodinium* diversity and ecology that have been revealed by more variable markers (psbA^ncr^) and microsatellites [@Finney:2010p8324; @Thornhill:2013dx]; namely, that clades B and C in the Caribbean comprise numerous different *Symbiodinium* species that tend to associate with different coral host taxa. Indeed, only the within-sample clustering approach assigned different dominant clade C OTUs to samples of *O. annularis* and *S. siderea* (Fig. 2), which have been differentiated based on other markers and microsatellites [@Thornhill:2013dx]. Therefore, when analyzed using within-sample clustering, ITS2 metabarcoding data may generate a good approximation of species-level diversity in most cases. In theory, this approach should also reflect patterns that would be observed by sequencing dominant ITS2 bands from denaturing gradient gel electrophoresis (DGGE; the most commonly use method of describing *Symbioinium* diversity prior to metabarcoding), since both methods rely on numerically dominant sequence variants to assign taxonomy. Furthermore, metabarcoding overcomes the primary limitations of the DGGE method by providing more quantitative data and highly sensitive detection of low abundance taxa [@Quigley:2014dv]. Therefore, we recommend a within-sample clustering approach for metabarcoding studies where many different *Symbiodinium* types are likely to be encountered.
  
  Importantly, the assumptions underlying a within-sample clustering approach -- that samples typically contain one *Symbiodinium* species that can be diagnosed by its most abundant intragenomic ITS2 variant -- will not always be met. Indeed, multiple *Symbiodinium* frequently co-occur in single coral colonies [@Silverstein:2012ub], which undermines support for assuming that variation within a sample is intragenomic. However, co-occurring *Symbiodinium* in many cases are members of different clades, whose ITS2 sequences are divergent enough to be resolved separately by 97% clustering. Thus, only when very closely related types co-occur in a sample [e.g., @Sampayo:2007p4020; @Wham:2016gl] is this approach likely to fail. Other problematic cases may occur when multiple intragenomic sequence variants are co-dominant (i.e., comprise nearly equal proportions of the rDNA array), such that slight differences in their relative abundance may lead to different OTU assignments in samples with otherwise highly similar sequence assemblages (e.g., Fig. SX). These challenges may be overcome by basing taxonomic assignment on more complex criteria than just the most abundant sequence variant, such as multiple co-dominant variants, or more evolutionarily derived variants.
  
  While no treatment of ITS2 data can provide species-level descriptions of *Symbiodinium* communities (since the marker itself does not provide this resolution), within-sample clustering nevertheless accommodates known evolutionary complexities in the ITS2 locus to provide community descriptions that better reflect species-level diversity than either across-sample clustering or no clustering at all. When treated with an appropriate bioinformatic approach, ITS2 metabarcoding can provide a comprehensive and quantitative analysis of *Symbiodinium* diversity, and can be easily applied across host species to rapidly survey the *Symbiodinium* metacommunity on a reef-wide scale.

#### *Symbiodinium* metacommunity ecology in St. John

  In the ten host species sampled in St. John, the most prevalent *Symbiodinium* were members of clades B and C, which is consistent with previous analyses of *Symbiodinium* diversity on shallow reefs in this region [@LaJeunesse:2004p122; @Correa:2009p4378; @Finney:2010p8324]. Less frequent associations between hosts and members of clades A and D were observed, although clade A dominated *P. astreoides* and some *P. furcata* colonies, and clade D dominated one *S. siderea* colony. Finally, clade G was observed at low relative abundances (<`r 100*round(clademax[5,2], 2)`%) in some samples, which likely would not have been detected without high-throughput sequencing. 
  
  While the environmental conditions differed between the north and south shores of St. John (e.g., wave exposure, temperature, and chlorophyll a), there were no statistically significant differences in symbiont community composition in the corals sampled on either shore. This may partly be due to small sample sizes, since two species showed a trend for differences between shores: *S. siderea* and *P. furcata* were more frequently dominated by *Symbiodinium* in clades D and A (respectively) on the south shore, compared to the north shore (Fig. 2). *Symbiodinium* in clades D and A are typically associated with warm and variable temperature regimes [e.g., @Baker:2004p21; @Oliver:2011cr] and shallow habitats with high light intensities [@LaJeunesse:2002p3917]. Therefore, it is consistent with expectations that they were more prevalent on the south shore, where temperatures were slightly higher, and chlorophyll levels were lower, suggesting greater light penetration. However, the lack of differentiation between shores in the symbiont communities of most host species suggests that environmental differences at this scale are not important drivers of community structure.
  
  Network analysis of associations between *Symbiodinium* types and coral species revealed metacommunity-level patterns in *Symbiodinium* ecology. First, ITS2 type B1 was "abundant" (i.e. >1% relative abundance in a sample) in seven of ten host species (Fig. 4), and "dominant" (i.e. >50%) in five of ten (Fig. 5), suggesting it is a generalist symbiont. However, analysis of other markers such as chloroplast and microsatellite loci has revealed that *Symbiodinium* ITS2 type B1 in the Caribbean is comprised of multiple lineages that show high host species fidelity [@Santos:2004p4044; @Finney:2010p8324; @Parkinson:2015gt]. Likewise, the apparent generalists *Symbiodinium* C3 (which dominated colonies of *S. siderea* and *M. cavernosa*) and *Symbiodinium* A4 (which dominated *P. astreoides* and *P. furcata*, Fig. 5) may also be comprised of multiple host-specialized lineages [@Thornhill:2013dx], which could be revealed using higher resolution genetic markers.
  
  In addition to hosting apparent generalist ITS2 lineages, most coral species sampled in St. John (except for *M. cavernosa* and *P. astreoides*) hosted abundant *Symbiodinium* OTUs that were not abundant in any other corals, and are thus apparently more host-specific. Unique OTUs in the B1-radiation were abundant in *M. alcicornis* and *D. cylindrus*, while others from both the B1- and B19-radiations [see @LaJeunesse:2004p122] were abundant in *F. fragum*, *S. radians*, and *P. strigosa*. Several clade C OTUs similarly were only abundant in one species: C1144 in *P. strigosa*, C7 in *O. annularis*, C1 in *S. siderea*, C1c/C45 and C45a in *P. furcata*, and C46 in *S. radians*. Interestingly, *S. radians* was the only species in which none of the abundant symbionts occurred in any other host, which may reflect the unique ecology of *S. radians* as the only study species that typically forms small, encrusting colonies as adults (XXX).
  
  To reveal symbionts that consistently occupied a 'background' niche, we identified OTUs present at <1% relative abundance in samples of three or more host species. This distribution suggests that these symbionts can occupy a range of hosts, but are unlikely to be dominant symbionts, at least under prevailing environmental conditions. The most prevalent 'background' symbiont in St. John was a member of clade D (*Symbiodinium trenchii* [@LaJeunesse:2014io]), which has been shown to proliferate within hosts during and after thermal stress [@Thornhill:2006p3638; @LaJeunesse:2009p7364; @Silverstein:2015er], or in marginal environments [@LaJeunesse:2010p7790]. Other symbionts occupying a background niche in St. John, such as *Symbiodinium* A4 and C3, may similarly be able to achieve dominance under different sets of environmental conditions. These background symbionts, therefore, may perform an important functional role within the metacommunity by broadening the fundamental niche space that a host may occupy (*sensu* @Bruno:2003p148), thereby increasing its resilience to environmental change [@Correa:2010p7787].
  
  Indeed, the ability of corals to associate with different symbionts may be an important trait that mediates their sensitivity to stress [@Putnam:2012wv], and their ability to change symbionts over time in response to environmental change [@Baker:2003p200]. While flexibility in symbiosis ecology has been quantified previously for individual corals (i.e., based on the diversity of symbionts co-occurring within a single host colony [@Putnam:2012wv]), here we quantify flexibility at the host species level based on the variability of symbiont community structure among multiple colonies (Fig. 6), a metric of beta diversity that can be statistically compared among host species [@Anderson:2006fs]. This metric revealed that *M. cavernosa*, *P. astreoides*, *S. radians*, and *D. cylindrus* had the lowest symbiotic flexibility, meaning that all sampled individuals had similar symbiont community structure. In these host species, symbiont communities may be more constrained by host biology and/or less responsive to environmental variation. On the other hand, *P. strigosa*, *P. furcata*, *M. alcicornis*, and *S. siderea* had high flexibility, meaning that sampled individuals displayed more divergent symbiont communities. Community structure in these hosts may be more responsive to variability in the environment [e.g., @Kennedy:2016do], or subject to greater stochasticity. While future work should investigate whether the scope for symbiont community change over time within individuals is linked to variability among individuals, we suggest that the latter represents a useful metric of symbiosis flexibility that can be easily quantified using metabarcoding data.

#### Conclusions

  ITS2 metabarcoding and within-sample OTU clustering represents a powerful approach to quantitatively and comprehensively describe *Symbiodinium* metacommunity composition on coral reefs. The scale and resolution of datasets generated in this way facilitate new analytical applications that can address critical topics in coral symbiosis ecology, including changes in dominant and background symbionts across environmental gradients and over time, and the role of metacommunity processes in shaping *Symbiodinium* communities. Describing these trends has the potential to greatly advance understanding of coral responses to environmental change.


### Acknowledgements

  Funding for this work was provided in part by the Long Term Research in Environmental Biology program of the US National Science Foundation (NSF) (DEB 13-50146). RC was supported by a NSF Postdoctoral Research Fellowship in Biology (NSF-PRFB 1400787). Remote sensing observations of sea surface temperature and phytoplankton pigment concentration were processed at the Institute for Marine Remote Sensing at the College of Marine Science, University of South Florida, in St. Petersburg, Florida, and provided courtesy of Frank Müller-Karger.

### References
