---
title: "*Symbiodinium* communities in reef corals of St. John, U.S.V.I. (2012)"
subtitle: "Data analysis report"
author: "Ross Cunning"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: 
  html_document: 
    toc: yes
    toc_depth: 2
    toc_float:
      collapsed: true
    code_folding: hide
---

```{r setup, include=FALSE}
knitr::opts_knit$set(root.dir = normalizePath(".."))
knitr::opts_chunk$set(echo = TRUE)
options(scipen = 3, digits = 9)
```

```{r setup_2, include=FALSE}
# Load all R scripts in "R/" directory
sapply(list.files(path="R", pattern="*.R", full.names=TRUE), source, .GlobalEnv)
# Load package libraries
library(phyloseq); library(vegan); library(multcompView); library(reshape2); library(igraph); library(stringr)
# Set colors for plotting clades
taxcolors <- matrix(c("#8dd3c7", "#ffffb3", "#bebada", "#fb8072", "#80b1d3", "#fdb462"), 
                    dimnames=list(c("CladeA", "CladeB", "CladeC", "CladeD", "CladeF", "CladeG")))
```

# Project
  This project investigates *Symbiodinium* communities in scleractinian and milleporine corals sampled at various sites on the north and south shores of St. John, US Virgin Islands in August 2012. The ITS2 region of nrDNA isolated from coral samples was amplified and sequenced by paired-end 300 bp reads on an Illumina MiSeq platform. Sequence data is processed through a custom bioinformatic pipeline and resulting OTU tables are statistically analyzed to ask questions regarding diversity and community structure of *Symbiodinium* within and across coral species and between shores. 
  The major aims of this work are to:
  
1. characterize *Symbiodinium* communities in a range of coral species using high-throughput sequencing (many for the first time)
2. forward novel bioinformatic (within-sample clustering) and statistical (beta diversity/distance to centroid, network analysis) approaches for the use of ITS2 data in *Symbiodinium* ecology. 

```{r STJmap, echo=F, message=F, cache=T, fig.height=3, fig.width=4, fig.align="center", fig.cap="Sampling locations in St. John"}
source("R/STJ_map.R")
```

# Bioinformatics
  A custom pipeline is used here to process ITS2 sequence data. Briefly, paired reads were merged using [illumina-utils](http://www.github.com/meren/illumina-utils) and filtered to keep only those with 3 mismatches or less. Chimeras were then removed by [usearch](http://www.drive5.com/usearch), and singletons were removed. From here, sequence data was processed using each of three approaches:

1. __100% OTU clustering.__ All ITS2 sequences from all samples were clustered at 100% identity using *uclust* and assigned taxonomy by BLAST to the [custom reference database](../data/ITS2db_trimmed.fasta).
2. __97% OTU clustering.__ All ITS2 sequences from all samples were clustered at 97% identity using *uclust* and the most abundant sequence within each OTU was selected as the representative sequence and assigned taxonomy by BLAST to the [custom reference database](../data/ITS2db_trimmed.fasta).
3. __97% OTU clustering within samples.__ Sequences from each individual sample were clustered separately at 97% similarity and the most abundant sequence from each cluster was selected as the representative sequence and assigned taxonomy by BLAST to the [custom reference database](../data/ITS2db_trimmed.fasta). OTUs with identical representative sequences were then merged across samples to create a global OTU table. By clustering at 97% _within_ samples, as opposed to across the entire dataset, OTUs from different samples that have _different representative sequences_ (i.e., different most abundant sequence within the 97% cluster) will remain separate even if these sequences are >97% similar to each other. This approach is used because _Symbiodinium_ ITS2 sequences are know to be multi-copy and intragenomically variable, and distinct _Symbiodinium_ types may contain some of the same ITS2 sequence variants in different relative proportions. Therefore, selecting the most abundant sequence variant _within_ a sample prevents a potentially distinct _Symbiodinium_ type from being collapsed into an OTU with a different representative sequence that happens to be more abundant in other samples within the dataset. For more discussion on this approach, see [this entry in my lab notebook](http://jrcunning.github.io/labnotebook//2016/02/12/ITS2_clustering.html).

OTU tables and taxonomic assignments from each of these bioinformatic approaches are used in comparative downstream analyses.

# Data pre-processing
### Import dataset
The [phyloseq](https://joey711.github.io/phyloseq/) package is used here to combine the OTU table, taxonomic assignments, and sample metadata into a single R data object (class 'phyloseq') to facilitate downstream analyses.
```{r data_import, cache=T}

```

```{r filter_taxa}
# Import taxonomic assignment data from local and global alignments
taxcolnames <- c("OTU", "BLAST.hit", "BLAST.pct.id", "BLAST.align.length", "BLAST.mismatches", "BLAST.gapopen", 
                 "BLAST.q.start", "BLAST.q.end", "BLAST.s.start", "BLAST.s.end", "BLAST.e.val", "BLAST.bit.score", 
                 "needle.hit", "needle.length", "needle.score")

TAX97 <- read.table("data/otus_97/tax_results.txt", sep=" ", col.names=taxcolnames)
TAX97$needle.score <- as.numeric(gsub("[^0-9.]", "", TAX97$needle.score))

TAX97bs <- read.table("data/otus_97_bysample/tax_results.txt", sep=" ", col.names=taxcolnames)
TAX97bs$needle.score <- as.numeric(gsub("[^0-9.]", "", TAX97bs$needle.score))
       
TAX97bsp <- read.table("data/otus_97_byspecies/tax_results.txt", sep=" ", col.names=taxcolnames)
TAX97bsp$needle.score <- as.numeric(gsub("[^0-9.]", "", TAX97bsp$needle.score))

TAX100 <- read.table("data/otus_100/tax_results.txt", sep=" ", col.names=taxcolnames)
TAX100$needle.score <- as.numeric(gsub("[^0-9.]", "", TAX100$needle.score))

# Merge identical taxonomic assignments (because some reference sequences are not unique...)
ident <- readLines("data/ITS2db_trimmed_notuniques_otus.txt")
ident <- strsplit(ident, split="\t")
names(ident) <- c("C3_multiple", "B19_multiple", "H1_multiple", "C33_multiple", "H5_multiple", "A2_multiple",
                  "B21/B38_multiple", "C35/C26_multiple", "A1/A13_multiple", "F5.2_multiple", "D1_multiple", 
                  "A10_multiple", "B2_multiple", "A1.2_multiple", "C1c/C45_multiple", "B1_multiple",
                  "C91_multiple", "C82_multiple", "D1a_multiple", "A4a_multiple", "C1_multiple",
                  "F5.2d_multiple", "C21/C32_multiple", "A3/A6_multiple", "C26/C35a_multiple")

merge.identical.hits <- function(x) {
  sapply(as.character(x), FUN=function(y) {
    ifelse(y %in% unlist(ident), 
           gsub("[0-9]*$", "", names(unlist(ident))[match(y, unlist(ident))]), 
           y)
    })
}

TAX97$BLAST.matched <- merge.identical.hits(TAX97$BLAST.hit)
TAX97$needle.matched <- merge.identical.hits(TAX97$needle.hit)

# OTUs for which BLAST and needle gave different assignments
nomatch <- subset(TAX97, BLAST.matched!=needle.matched)
nomatch


# Get percent identity of global alignment to needle hits
# import reference sequences
ref <- readDNAStringSet("data/ITS2db_trimmed.fasta")
# import otu sequences
otus97 <- readDNAStringSet("data/otus_97/97_otus_rep_set.fasta")
names(otus97) <- gsub(" .*$", "", names(otus97))
# align each otu sequence to its best needle hit, using same parameters as command line needle
# "global-local" forces whole reference/pattern sequence to be used, but subject/query can be a concescutive subsequence -- this is because queries may include slightly longer region of gene than references because references were trimmed by o-smart-trim. this means that query sequence can have more bases than the reference at beginning or end of the sequence (=end gaps in alignment) without penalties.
EDNAFULL <- as.matrix(read.table("data/EDNAFULL", skip=8, header=T, row.names=1))
n.psa <- pairwiseAlignment(subject=otus97[TAX97$OTU], pattern=ref[TAX97$needle.hit], type="global-local",
                           substitutionMatrix=EDNAFULL,
                           gapOpening=10.0, gapExtension=0.5)

# calculate percent similarity to reference with indels counting as single difference regardless of length
TAX97 <- within(TAX97, {
  needle.hit.length <- width(pattern(n.psa))
  needle.mismatches <- nmismatch(n.psa)
  needle.insertions <- unlist(lapply(as.list(insertion(n.psa)), length))
  needle.deletions <- unlist(lapply(as.list(deletion(n.psa)), length))
  needle.similarity <- (needle.hit.length-needle.mismatches-needle.insertions-needle.deletions)/needle.hit.length
})

g.psa <- pairwiseAlignment(subject=otus97[TAX97$OTU], pattern=ref[TAX97$needle.hit], type="global",
                           substitutionMatrix=EDNAFULL,
                           gapOpening=10.0, gapExtension=0.5)

# calculate percent similarity to reference with indels counting as single difference regardless of length
TAX97 <- within(TAX97, {
  gneedle.hit.length <- width(pattern(g.psa))
  gneedle.mismatches <- nmismatch(g.psa)
  gneedle.insertions <- unlist(lapply(as.list(insertion(g.psa)), length))
  gneedle.deletions <- unlist(lapply(as.list(deletion(g.psa)), length))
  gneedle.similarity <- (gneedle.hit.length-gneedle.mismatches-gneedle.insertions-gneedle.deletions)/gneedle.hit.length
})


aligned(n.psa[])
n.psa
aligned(n.psa[1])
TAX97$needle.hit.length <- width(pattern(n.psa))
TAX97$needle.mismatches <- nmismatch(n.psa)
TAX97$needle.insertions <- unlist(lapply(as.list(insertion(n.psa)), length))
TAX97$needle.deletions <- unlist(lapply(as.list(deletion(n.psa)), length))
TAX97$needle.similarity <- with(TAX97, (needle.hit.length - needle.mismatches - needle.insertions - needle.deletions) / needle.hit.length)

with(TAX97[115, ], {
  aln <<- pairwiseAlignment(subject=otus97[OTU], pattern=ref[needle.hit], type="global-local",
                            substitutionMatrix=EDNAFULL,
                            gapOpening=10.0, gapExtension=0.5)
})
aln

aln <- pairwiseAlignment(subject="ACTTCTCTATCTATCTTGGTACG", pattern="AATTCTCGGTACG", type="global")
aln2 <- pairwiseAlignment(subject="ACTTCTACG", pattern="AATTCTCGGTACG", type="global")
aln
aln2
width(aligned(aln))
nmatch(aln)/width(aligned(aln))


# calculate pct id of all positions relative to reference/(pattern)
nmatch(aln)/width(pattern(aln))
# calculate pct id counting indels as single difference regardless of length = GAST definition
(width(pattern(aln)) - nmismatch(aln) - length(unlist(insertion(aln))) - length(unlist(deletion(aln))))/width(pattern(aln))

stringDist(aln)

nmatch(aln)
nmismatch(aln)
width(nindel(aln))
length(unlist(insertion(aln))); length(unlist(deletion(aln)))
length(unlist(insertion(aln)))
length(unlist(deletion(aln)))
insertion(aln)
deletion(aln)
unlist(nindel(aln))
nindel(aln)$insertion
str(nindel(aln))
b <- n.psa[11]
as.list(indel(subject(b)))
as.list(indel(subject(aln)))
length(as.data.frame(deletion(b))[[1]])
indel(b)
length(unlist(insertion(b)))



class(EDNAFULL)
n.psa[47]
summary(n.psa[47])
nmatch(n.psa[47])
nindel(n.psa[47])
nmismatch(n.psa[47])
nchar(pattern(n.psa[2]))
246/255
nmismatch(n.psa[2])

```

```{r build_phyloseq}
# Import OTU tables
OTU97 <- otu_table(read.table("data/otus_97/97_otus.tsv", header=T, check.names=F, row.names=1,
                              skip=1, sep="\t", comment.char=""), taxa_are_rows=T)
OTU97bs <- otu_table(read.table("data/otus_97_bysample/97_otus_bysample.tsv", header=T, check.names=F, row.names=1,
                              skip=1, sep="\t", comment.char=""), taxa_are_rows=T)
#OTU97bsp <- otu_table(read.table("data/97_otus_byspecies.tsv", header=T, check.names=F, row.names=1,
#                              skip=1, sep="\t", comment.char=""), taxa_are_rows=T)
OTU100 <- otu_table(read.table("data/otus_100/100_otus.tsv", header=T, check.names=F, row.names=1,
                               skip=1, sep="\t", comment.char=""), taxa_are_rows=T)
# Import taxonomy

# Import sample data
SAM <- sample_data(read.delim("data/mapping_file.txt", sep="\t", header=T, row.names=1))
# Build phyloseq objects
phy97 <- phyloseq(OTU97, TAX97, SAM)
phy97bs <- phyloseq(OTU97bs, TAX97bs, SAM)
phy97bsp <- phyloseq(OTU97bsp, TAX97bsp, SAM)
phy100 <- phyloseq(OTU100, TAX100, SAM)
```

### Filter dataset
```{r filter_dataset, cache=T}





# Filter OTUs by minimum count
# Set threshold count
n <- 10
# Identify OTUs below threshold count
taxa97 <- taxa_sums(phy97)[which(taxa_sums(phy97) >= n)]
taxa97bs <- taxa_sums(phy97bs)[which(taxa_sums(phy97bs) >= n)]
taxa97bsp <- taxa_sums(phy97bsp)[which(taxa_sums(phy97bsp) >= n)] 
taxa100 <- taxa_sums(phy100)[which(taxa_sums(phy100) >= n)]
# Remove taxa below threshold count
phy97.f <- prune_taxa(names(taxa97), phy97)
phy97bs.f <- prune_taxa(names(taxa97bs), phy97bs)
phy97bsp.f <- prune_taxa(names(taxa97bsp), phy97bsp)
phy100.f <- prune_taxa(names(taxa100), phy100)

# Filter samples by minimum count
# Set threshold number of reads
sn <- 200
# Remove samples with fewer reads than threshold
phy97.f <- prune_samples(sample_sums(phy97.f)>=sn, phy97.f)
phy97bs.f <- prune_samples(sample_sums(phy97bs.f)>=sn, phy97bs.f)
phy97bsp.f <- prune_samples(sample_sums(phy97bsp.f)>=sn, phy97bsp.f)
phy100.f <- prune_samples(sample_sums(phy100.f)>=sn, phy100.f)

# Filter OTUs by minimum count again in case any dropped below threshold after filtering samples
# Identify OTUs below threshold count
taxa97 <- taxa_sums(phy97.f)[which(taxa_sums(phy97.f) >= n)]
taxa97bs <- taxa_sums(phy97bs.f)[which(taxa_sums(phy97bs.f) >= n)]
taxa97bsp <- taxa_sums(phy97bsp.f)[which(taxa_sums(phy97bsp.f) >= n)] 
taxa100 <- taxa_sums(phy100.f)[which(taxa_sums(phy100.f) >= n)]
# Remove taxa below threshold count
phy97.f <- prune_taxa(names(taxa97), phy97.f)
phy97bs.f <- prune_taxa(names(taxa97bs), phy97bs.f)
phy97bsp.f <- prune_taxa(names(taxa97bsp), phy97bsp.f)
phy100.f <- prune_taxa(names(taxa100), phy100.f)
```
* Minimum count to retain OTU: `r n`
* Minimum count to retain sample: `r sn`

### Descriptive statistics {.tabset}

#### Filtered dataset
```{r phy.f_histograms, cache=T, fig.height=4, fig.width=8}
# Compute summary statistics
stats97.f <- data.frame(`97% OTUs`=t(phystats(phy97.f)), check.names=F)
stats97bs.f <- data.frame(`97% within-sample OTUs`=t(phystats(phy97bs.f)), check.names=F)
stats97bsp.f <- data.frame(`97% within-species OTUs`=t(phystats(phy97bsp.f)), check.names=F)
stats100.f <- data.frame(`100% OTUs`=t(phystats(phy100.f)), check.names=F)

# Create and plot histograms
taxhist97 <- hist(log10(taxa_sums(phy97.f)), plot=F)
taxhist97bs <- hist(log10(taxa_sums(phy97bs.f)), plot=F)
taxhist97bsp <- hist(log10(taxa_sums(phy97bsp.f)), plot=F)
taxhist100 <- hist(log10(taxa_sums(phy100.f)), plot=F)
samhist97 <- hist(log10(sample_sums(phy97.f)), plot=F)
samhist97bs <- hist(log10(sample_sums(phy97bs.f)), plot=F)
samhist97bsp <- hist(log10(sample_sums(phy97bsp.f)), plot=F)
samhist100 <- hist(log10(sample_sums(phy100.f)), plot=F)

par(mfrow=c(2, 4), mar=c(3,3,1,1))
plot(taxhist97, col="black", main="97% OTU counts", xlim=c(0, 6), las=1, mgp=c(2,0.5,0),
     xlab="No. sequences (log10)", ylab="No. OTUs", cex.lab=0.75, cex.axis=0.75)
plot(taxhist97bs, col="black", main="97% sample OTU counts", xlim=c(0, 6), las=1, mgp=c(2,0.5,0),
     xlab="No. sequences (log10)", ylab="No. OTUs", cex.lab=0.75, cex.axis=0.75)
plot(taxhist97bsp, col="black", main="97% species OTU counts", xlim=c(0, 6), las=1, mgp=c(2,0.5,0),
     xlab="No. sequences (log10)", ylab="No. OTUs", cex.lab=0.75, cex.axis=0.75)
plot(taxhist100, col="black", main="100% OTU counts", xlim=c(0, 6), las=1, mgp=c(2,0.5,0),
     xlab="No. sequences (log10)", ylab="No. OTUs", cex.lab=0.75, cex.axis=0.75)
plot(samhist97, col="black", main="97% OTU reads per sample", xlim=c(0, 6), las=1, mgp=c(2,0.5,0),
     xlab="No. sequences (log10)", ylab="No. samples", cex.lab=0.75, cex.axis=0.75)
plot(samhist97bs, col="black", main="97% sample OTU reads per sample", xlim=c(0, 6), las=1, mgp=c(2,0.5,0),
     xlab="No. sequences (log10)", ylab="No. samples", cex.lab=0.75, cex.axis=0.75)
plot(samhist97bsp, col="black", main="97% species OTU reads per sample", xlim=c(0, 6), las=1, mgp=c(2,0.5,0),
     xlab="No. sequences (log10)", ylab="No. samples", cex.lab=0.75, cex.axis=0.75)
plot(samhist100, col="black", main="100% OTU reads per sample", xlim=c(0, 6), las=1, mgp=c(2,0.5,0),
     xlab="No. sequences (log10)", ylab="No. samples", cex.lab=0.75, cex.axis=0.75)

# Create stats table
knitr::kable(cbind(stats97.f, stats97bs.f, stats97bsp.f, stats100.f))
```

#### Raw dataset
```{r phy_stats, cache=T, fig.height=4, fig.width=8}
# Compute summary statistics
stats97 <- data.frame(`97% OTUs`=t(phystats(phy97)), check.names=F)
stats97bs <- data.frame(`97% within-sample OTUs`=t(phystats(phy97bs)), check.names=F)
stats97bsp <- data.frame(`97% within-species OTUs`=t(phystats(phy97bsp)), check.names=F)
stats100 <- data.frame(`100% OTUs`=t(phystats(phy100)), check.names=F)

# Create and plot histograms
taxhist97 <- hist(log10(taxa_sums(phy97)), plot=F)
samhist97 <- hist(log10(sample_sums(phy97)), plot=F)

taxhist97bs <- hist(log10(taxa_sums(phy97bs)), plot=F)
samhist97bs <- hist(log10(sample_sums(phy97bs)), plot=F)

taxhist97bsp <- hist(log10(taxa_sums(phy97bsp)), plot=F)
samhist97bsp <- hist(log10(sample_sums(phy97bsp)), plot=F)

taxhist100 <- hist(log10(taxa_sums(phy100)), plot=F)
samhist100 <- hist(log10(sample_sums(phy100)), plot=F)

par(mfrow=c(2, 4), mar=c(3,3,1,1))
plot(taxhist97, col="black", main="97% OTU counts", xlim=c(0, 6), las=1, mgp=c(2,0.5,0),
     xlab="No. sequences (log10)", ylab="No. OTUs", cex.lab=0.75, cex.axis=0.75)
plot(taxhist97bs, col="black", main="97% within-sample OTU counts", xlim=c(0, 6), las=1, mgp=c(2,0.5,0),
     xlab="No. sequences (log10)", ylab="No. OTUs", cex.lab=0.75, cex.axis=0.75)
plot(taxhist97bsp, col="black", main="97% within-species OTU counts", xlim=c(0, 6), las=1, mgp=c(2,0.5,0),
     xlab="No. sequences (log10)", ylab="No. OTUs", cex.lab=0.75, cex.axis=0.75)
plot(taxhist100, col="black", main="100% OTU counts", xlim=c(0, 6), las=1, mgp=c(2,0.5,0),
     xlab="No. sequences (log10)", ylab="No. OTUs", cex.lab=0.75, cex.axis=0.75)
plot(samhist97, col="black", main="97% OTU reads per sample", xlim=c(0, 6), las=1, mgp=c(2,0.5,0),
     xlab="No. sequences (log10)", ylab="No. samples", cex.lab=0.75, cex.axis=0.75)
plot(samhist97bs, col="black", main="97% within-sample OTU reads per sample", xlim=c(0, 6), las=1, mgp=c(2,0.5,0),
     xlab="No. sequences (log10)", ylab="No. samples", cex.lab=0.75, cex.axis=0.75)
plot(samhist97bs, col="black", main="97% within-species OTU reads per sample", xlim=c(0, 6), las=1, mgp=c(2,0.5,0),
     xlab="No. sequences (log10)", ylab="No. samples", cex.lab=0.75, cex.axis=0.75)
plot(samhist100, col="black", main="100% OTU reads per sample", xlim=c(0, 6), las=1, mgp=c(2,0.5,0),
     xlab="No. sequences (log10)", ylab="No. samples", cex.lab=0.75, cex.axis=0.75)

# Create stats table
knitr::kable(cbind(stats97, stats97bs, stats97bsp, stats100))
```

### Transform count data
Count data are transformed to both relative abundance (proportions) and square-root proportions for downstream statistical analyses.
```{r transform}
# Convert to proportion (relative abundance)
phy97.f.p <- transform_sample_counts(phy97.f, function(x) x/sum(x))
phy97bs.f.p <- transform_sample_counts(phy97bs.f, function(x) x/sum(x))
phy97bsp.f.p <- transform_sample_counts(phy97bsp.f, function(x) x/sum(x))
phy100.f.p <- transform_sample_counts(phy100.f, function(x) x/sum(x))
# Apply transformation function
transform <- function(x) sqrt(x/sum(x))  # Set transformation function
phy97.f.t <- transform_sample_counts(phy97.f, transform)  # Transform data
phy97bs.f.t <- transform_sample_counts(phy97bs.f, transform) 
phy97bsp.f.t <- transform_sample_counts(phy97bsp.f, transform) 
phy100.f.t <- transform_sample_counts(phy100.f, transform)
```

# Clade overview
Here the composition of each sample is plotted as a horizontal bar, sorted by species and shore. Each segment of the bar represents an individual OTU and is colored by clade. This plot is a nice overview of the whole dataset but only provides coarse information at the clade level.
```{r cladecomp97bs, cache=T, fig.height=6, fig.width=4}
cladeAbund <- aggregate(data.frame(RelAbund=rowSums(otu_table(phy97bs.f.p))),
                        by=list(Clade=data.frame(tax_table(phy97bs.f.p))$Clade), FUN=sum)
sum(cladeAbund$RelAbund)
cladeAbund$Prop <- prop.table(cladeAbund$RelAbund)
pie(cladeAbund$Prop, col=taxcolors)
dotchart(cladeAbund$Prop)
bars <- barplot(cladeAbund$Prop*100, col=taxcolors, space=0,
                names.arg=cladeAbund$Clade, xlab=expression(paste(italic('Symbiodinium'), " Clade")),
                ylab="Relative abundance (%)")
text(bars, cladeAbund$Prop*100+2, labels=round(cladeAbund$Prop*100, 1), xpd=T)

cladeAbund$Notus <- table(data.frame(tax_table(phy97bs.f.p))$Clade)
cladeAbund


par(mfrow=c(1,1), mar=c(2, 1.5, 2, 5), lwd=0.1, cex=0.7)
# Plot composition of 97% within-sample OTUs colored by clade
composition(phy97bs.f.p, col=taxcolors[factor(data.frame(tax_table(phy97bs.f.p))[order(data.frame(tax_table(phy97bs.f.p))$Subtype), ]$Clade, levels=c("A","B","C","D","F","G"))], legend=T)
# Plot composition of 97% within-sample OTUs colored by OTU
#composition(phy97bs.f.p, col=rainbow(ntaxa(phy97bs.f.p)), legend=F)

```

-----

# *Symbiodinium* in each coral
For each coral species, barplots are presented showing the relative abundance of OTUs obtained by 100%, 97%, and 97%-within-sample clustering. OTUs comprising more than 4% of a sample are labeled with the unique OTU number and the *Symbiodinium* subtype and NCBI GenBank accession number of the closest BLAST hit for that OTU in the reference database. OTU numbers and barplot colors are NOT comparable across the three clustering methods.

## *Diploria strigosa*  
Notice how 97% OTUs and 97% within-sample OTUs are **NOT THE SAME!** Specifically, notice how in the 97% OTU dataset, all samples except two are dominated by the same OTU, denovo235. In the 97% within-sample OTU dataset, those samples are dominated by three different OTUs -- denovo73, denovo231, and denovo10. Look at the 100% OTU dataset to see the composition of unique sequence variants in these samples. Indeed, these samples are dominated by **different sequence variants**. However, in the 97% OTU approach, they have all been collapsed into the same OTU, while in the 97% within-sample clustering approach, samples with different dominant sequence variants are maintained as separate OTUs, even though these different sequence variants are more than 97% similar to each other. I suggest that the  within-sample clustering approach is a better way of treating *Symbiodinium* ITS2 sequence data.
```{r dstr, cache=T, fig.height=16, fig.width=8}
# Create subsetted phyloseq objects for Diploria strigosa
dstr97.f.p <- subset_samples(phy97.f.p, Species=="strigosa")
dstr97bs.f.p <- subset_samples(phy97bs.f.p, Species=="strigosa")
dstr97bsp.f.p <- subset_samples(phy97bsp.f.p, Species=="strigosa")
dstr100.f.p <- subset_samples(phy100.f.p, Species=="strigosa")
# Plot custom barplots for Diploria strigosa
par(mfrow=c(4,1), mar=c(4,4,2,0), mgp=c(0,0.5,0), lwd=0.1)
otubarplot(dstr97.f.p, main="97% OTUs")
otubarplot(dstr97bsp.f.p, main="97% within-species OTUs")
otubarplot(dstr97bs.f.p, main="97% within-sample OTUs")
otubarplot(dstr100.f.p, main="100% OTUs")
```

```{r dstr.net, cache=T, fig.height=6, fig.width=6}
dstr.net <- makenet(dstr97bs.f.p, 0)
set.seed(54538)
plotnet(dstr.net)
```

#### Network analysis for *D. strigosa*
#### Number of OTUs: `r table(is.na(V(dstr.net)$Clade))[[1]]`
White circles represent coral samples, colored circles represent *Symbiodinium* OTUs (97% within-sample). Thickness of lines corresponds to the relative abundance of an OTU within a sample. 

-----

## *Porites furcata*
```{r pfur, cache=T, fig.height=16, fig.width=8}
# Create subsetted phyloseq objects for Porites furcata
pfur97.f.p <- subset_samples(phy97.f.p, Species=="furcata")
pfur97bsp.f.p <- subset_samples(phy97bsp.f.p, Species=="furcata")
pfur97bs.f.p <- subset_samples(phy97bs.f.p, Species=="furcata")
pfur100.f.p <- subset_samples(phy100.f.p, Species=="furcata")
# Plot custom barplots for Porites furcata
par(mfrow=c(4,1), mar=c(4,4,2,0), mgp=c(0,0.5,0), lwd=0.1)
otubarplot(pfur97.f.p, main="97% OTUs")
otubarplot(pfur97bsp.f.p, main="97% within-species OTUs")
otubarplot(pfur97bs.f.p, main="97% within-sample OTUs")
otubarplot(pfur100.f.p, main="100% OTUs")
```
```{r pfur.net, cache=T, fig.height=6, fig.width=6}
# Network analysis
pfur.net <- makenet(pfur97bs.f.p, 0)
par(mar=c(0,0,0,0))
set.seed(54538)
plotnet(pfur.net)
```

#### Network analysis for *P. furcata*
#### Number of OTUs: `r table(is.na(V(pfur.net)$Clade))[[1]]`
White circles represent coral samples, colored circles represent *Symbiodinium* OTUs (97% within-sample). Thickness of lines corresponds to the relative abundance of an OTU within a sample. 

## *Orbicella annularis*
```{r oann, cache=F, fig.height=16, fig.width=8}
# Create subsetted phyloseq objects for Orbicella annularis
oann97.f.p <- subset_samples(phy97.f.p, Species=="annularis")
oann97bsp.f.p <- subset_samples(phy97bsp.f.p, Species=="annularis")
oann97bs.f.p <- subset_samples(phy97bs.f.p, Species=="annularis")
oann100.f.p <- subset_samples(phy100.f.p, Species=="annularis")
# Plot custom barplots for Orbicella annularis
par(mfrow=c(4,1), mar=c(4,4,2,0), mgp=c(0,0.5,0), lwd=0.1)
otubarplot(oann97.f.p, main="97% OTUs")
otubarplot(oann97bsp.f.p, main="97% within-species OTUs")
otubarplot(oann97bs.f.p, main="97% within-sample OTUs")
otubarplot(oann100.f.p, main="100% OTUs")
```
```{r oann.net, cache=T, fig.height=6, fig.width=6}
# Network analysis
oann.net <- makenet(oann97bs.f.p, 0)
par(mar=c(0,0,0,0))
set.seed(54538)
plotnet(oann.net)
```

#### Network analysis for *O. annularis*
#### Number of OTUs: `r table(is.na(V(oann.net)$Clade))[[1]]`
White circles represent coral samples, colored circles represent *Symbiodinium* OTUs (97% within-sample). Thickness of lines corresponds to the relative abundance of an OTU within a sample.

## *Millepora alcicornis*
```{r malc, cache=T, fig.height=16, fig.width=8}
# Create subsetted phyloseq objects for Millepora alcicornis
malc97.f.p <- subset_samples(phy97.f.p, Species=="alcicornis")
malc97bsp.f.p <- subset_samples(phy97bsp.f.p, Species=="alcicornis")
malc97bs.f.p <- subset_samples(phy97bs.f.p, Species=="alcicornis")
malc100.f.p <- subset_samples(phy100.f.p, Species=="alcicornis")
# Plot custom barplots for Millepora alcicornis
par(mfrow=c(4,1), mar=c(4,4,2,0), mgp=c(0,0.5,0), lwd=0.1)
otubarplot(malc97.f.p, main="97% OTUs")
otubarplot(malc97bsp.f.p, main="97% within-sample OTUs")
otubarplot(malc97bs.f.p, main="97% within-sample OTUs")
otubarplot(malc100.f.p, main="100% OTUs")
```
```{r malc.net, cache=T, fig.height=6, fig.width=6}
# Network analysis
malc.net <- makenet(malc97bs.f.p, 0)
par(mar=c(0,0,0,0))
set.seed(54538)
plotnet(malc.net)
```

#### Network analysis for *M. alcicornis*
#### Number of OTUs: `r table(is.na(V(malc.net)$Clade))[[1]]`
White circles represent coral samples, colored circles represent *Symbiodinium* OTUs (97% within-sample). Thickness of lines corresponds to the relative abundance of an OTU within a sample.

## *Siderastrea siderea*
```{r ssid, cache=T, fig.height=16, fig.width=8}
# Create subsetted phyloseq objects for Siderastrea siderea
ssid97.f.p <- subset_samples(phy97.f.p, Species=="siderea")
ssid97bsp.f.p <- subset_samples(phy97bsp.f.p, Species=="siderea")
ssid97bs.f.p <- subset_samples(phy97bs.f.p, Species=="siderea")
ssid100.f.p <- subset_samples(phy100.f.p, Species=="siderea")
# Plot custom barplots for Siderastrea siderea
par(mfrow=c(4,1), mar=c(4,4,2,0), mgp=c(0,0.5,0), lwd=0.1)
otubarplot(ssid97.f.p, main="97% OTUs")
otubarplot(ssid97bsp.f.p, main="97% within-species OTUs")
otubarplot(ssid97bs.f.p, main="97% within-sample OTUs")
otubarplot(ssid100.f.p, main="100% OTUs")
```
```{r ssid.net, cache=T, fig.height=6, fig.width=6}
# Network analysis
ssid.net <- makenet(ssid97bs.f.p, 0)
par(mar=c(0,0,0,0))
set.seed(834)
plotnet(ssid.net)
```

#### Network analysis for *S. siderea*
#### Number of OTUs: `r table(is.na(V(ssid.net)$Clade))[[1]]`
White circles represent coral samples, colored circles represent *Symbiodinium* OTUs (97% within-sample). Thickness of lines corresponds to the relative abundance of an OTU within a sample.

## *Favia fragum*
```{r ffra, cache=T, fig.height=16, fig.width=8}
# Create subsetted phyloseq objects for Favia fragum
ffra97.f.p <- subset_samples(phy97.f.p, Species=="fragum")
ffra97bsp.f.p <- subset_samples(phy97bsp.f.p, Species=="fragum")
ffra97bs.f.p <- subset_samples(phy97bs.f.p, Species=="fragum")
ffra100.f.p <- subset_samples(phy100.f.p, Species=="fragum")
# Plot custom barplots for Favia fragum
par(mfrow=c(4,1), mar=c(4,4,2,0), mgp=c(0,0.5,0), lwd=0.1)
otubarplot(ffra97.f.p, main="97% OTUs")
otubarplot(ffra97bsp.f.p, main="97% within-species OTUs")
otubarplot(ffra97bs.f.p, main="97% within-sample OTUs")
otubarplot(ffra100.f.p, main="100% OTUs")
```
```{r ffra.net, cache=T, fig.height=6, fig.width=6}
# Network analysis
ffra.net <- makenet(ffra97bs.f.p, 0)
par(mar=c(0,0,0,0))
set.seed(54538)
plotnet(ffra.net)
```

#### Network analysis for *F. fragum*
#### Number of OTUs: `r table(is.na(V(ffra.net)$Clade))[[1]]`
White circles represent coral samples, colored circles represent *Symbiodinium* OTUs (97% within-sample). Thickness of lines corresponds to the relative abundance of an OTU within a sample.

## *Siderastrea radians*
```{r srad, cache=T, fig.height=16, fig.width=8}
# Create subsetted phyloseq objects for Siderastrea radians
srad97.f.p <- subset_samples(phy97.f.p, Species=="radians")
srad97bsp.f.p <- subset_samples(phy97bsp.f.p, Species=="radians")
srad97bs.f.p <- subset_samples(phy97bs.f.p, Species=="radians")
srad100.f.p <- subset_samples(phy100.f.p, Species=="radians")
# Plot custom barplots for Siderastrea radians
par(mfrow=c(4,1), mar=c(4,4,2,0), mgp=c(0,0.5,0), lwd=0.1)
otubarplot(srad97.f.p, main="97% OTUs")
otubarplot(srad97bsp.f.p, main="97% within-species OTUs")
otubarplot(srad97bs.f.p, main="97% within-sample OTUs")
otubarplot(srad100.f.p, main="100% OTUs")
```
```{r srad.net, cache=T, fig.height=6, fig.width=6}
# Network analysis
srad.net <- makenet(srad97bs.f.p, 0)
par(mar=c(0,0,0,0))
set.seed(54538)
plotnet(srad.net)
```

#### Network analysis for *S. radians*
#### Number of OTUs: `r table(is.na(V(srad.net)$Clade))[[1]]`
White circles represent coral samples, colored circles represent *Symbiodinium* OTUs (97% within-sample). Thickness of lines corresponds to the relative abundance of an OTU within a sample.

## *Porites astreoides*
```{r past, cache=T, fig.height=16, fig.width=8}
# Create subsetted phyloseq objects for Porites astreoides
past97.f.p <- subset_samples(phy97.f.p, Species=="astreoides")
past97bsp.f.p <- subset_samples(phy97bsp.f.p, Species=="astreoides")
past97bs.f.p <- subset_samples(phy97bs.f.p, Species=="astreoides")
past100.f.p <- subset_samples(phy100.f.p, Species=="astreoides")
# Plot custom barplots for Porites astreoides
par(mfrow=c(4,1), mar=c(4,4,2,0), mgp=c(0,0.5,0), lwd=0.1)
otubarplot(past97.f.p, main="97% OTUs")
otubarplot(past97bsp.f.p, main="97% within-species OTUs")
otubarplot(past97bs.f.p, main="97% within-sample OTUs")
otubarplot(past100.f.p, main="100% OTUs")
```
```{r past.net, cache=T, fig.height=6, fig.width=6}
# Network analysis
past.net <- makenet(past97bs.f.p, 0)
par(mar=c(0,0,0,0))
set.seed(54538)
plotnet(past.net)
```

#### Network analysis for *P. astreoides*
#### Number of OTUs: `r table(is.na(V(past.net)$Clade))[[1]]`
White circles represent coral samples, colored circles represent *Symbiodinium* OTUs (97% within-sample). Thickness of lines corresponds to the relative abundance of an OTU within a sample.

## *Dendrogyra cylindrus*
```{r dcyl, cache=T, fig.height=16, fig.width=8}
# Create subsetted phyloseq objects for Dendrogyra cylindrus
dcyl97.f.p <- subset_samples(phy97.f.p, Species=="cylindrus")
dcyl97bsp.f.p <- subset_samples(phy97bsp.f.p, Species=="cylindrus")
dcyl97bs.f.p <- subset_samples(phy97bs.f.p, Species=="cylindrus")
dcyl100.f.p <- subset_samples(phy100.f.p, Species=="cylindrus")
# Plot custom barplots for Dendrogyra cylindrus
par(mfrow=c(4,1), mar=c(4,4,2,0), mgp=c(0,0.5,0), lwd=0.1)
otubarplot(dcyl97.f.p, main="97% OTUs")
otubarplot(dcyl97bsp.f.p, main="97% within-species OTUs")
otubarplot(dcyl97bs.f.p, main="97% within-sample OTUs")
otubarplot(dcyl100.f.p, main="100% OTUs")
```
```{r dcyl.net, cache=T, fig.height=6, fig.width=6}
# Network analysis
dcyl.net <- makenet(dcyl97bs.f.p, 0)
par(mar=c(0,0,0,0))
set.seed(54538)
plotnet(dcyl.net)
```

#### Network analysis for *D. cylindrus*
#### Number of OTUs: `r table(is.na(V(dcyl.net)$Clade))[[1]]`
White circles represent coral samples, colored circles represent *Symbiodinium* OTUs (97% within-sample). Thickness of lines corresponds to the relative abundance of an OTU within a sample.

## *Montastraea cavernosa*
```{r mcav, cache=T, fig.height=16, fig.width=8}
# Create subsetted phyloseq objects for Montastraea cavernosa
mcav97.f.p <- subset_samples(phy97.f.p, Species=="cavernosa")
mcav97bsp.f.p <- subset_samples(phy97bsp.f.p, Species=="cavernosa")
mcav97bs.f.p <- subset_samples(phy97bs.f.p, Species=="cavernosa")
mcav100.f.p <- subset_samples(phy100.f.p, Species=="cavernosa")
# Plot custom barplots for Montastraea cavernosa
par(mfrow=c(4,1), mar=c(4,4,2,0), mgp=c(0,0.5,0), lwd=0.1)
otubarplot(mcav97.f.p, main="97% OTUs")
otubarplot(mcav97bsp.f.p, main="97% within-species OTUs")
otubarplot(mcav97bs.f.p, main="97% within-sample OTUs")
otubarplot(mcav100.f.p, main="100% OTUs")
```
```{r mcav.net, cache=T, fig.height=6, fig.width=6}
# Network analysis
mcav.net <- makenet(mcav97bs.f.p, 0)
par(mar=c(0,0,0,0))
set.seed(54538)
plotnet(mcav.net)
```

#### Network analysis for *M. cavernosa*
#### Number of OTUs: `r table(is.na(V(mcav.net)$Clade))[[1]]`
White circles represent coral samples, colored circles represent *Symbiodinium* OTUs (97% within-sample). Thickness of lines corresponds to the relative abundance of an OTU within a sample.

# Network analyses
## "Resident" symbionts
#### (>1% relative abundance)
In this network, each coral species is collapsed into a single node (white squares), and connected to every *Symbiodinium* OTU (colored circles) that comprises **at least 1% relative abundance** within at least one individual of that species. Thus, very low abundance OTUs are not represented. The thickness of connections between coral species and OTUs is scaled by the relative proportion of samples of that species in which that OTU was present at ≥1% relative abundance (i.e., the frequency of presence at ≥1%). The size of *Symbiodinium* OTU nodes in the network is scaled by the number of species in which they occur, and colored according to clade.
```{r networks, fig.width=5, fig.height=5, fig.align="center"}
# Get network with one node for each species.
otudf <- data.frame(t(otu_table(phy97bs.f.p)), check.names=F)
# Percent of samples of species that contain OTU at more than 1% relative abundance
sp.ag <- aggregate(otudf, by=list(Species=data.frame(sample_data(phy97bs.f.p))$Species), 
                   FUN=function(x) length(x[x>0.01])/length(x))
rownames(sp.ag) <- sp.ag$Species
sp.ag <- sp.ag[, -1]
edges <- setNames(melt(as.matrix(sp.ag)), nm=c("source", "target", "weight"))  # Melt data
edges <- droplevels(edges[edges$weight>0, ])
# Create "nodes" table from sample and tax data
sdf <- data.frame(id=levels(data.frame(sample_data(phy97bs.f.p))$Species))
tdf <- data.frame(tax_table(phy97bs.f.p))[rownames(data.frame(tax_table(phy97bs.f.p))) %in% edges$target, ]
tdf$id <- rownames(tdf)
nodes <- merge(sdf, tdf, all=T)
nodes$type <- ifelse(is.na(nodes$Clade), 1, 2)
# Create network
net <- graph_from_data_frame(d=edges, vertices=nodes, directed=F)

# Best one for now ---------
V(net)$shape <- c("square", "circle")[factor(V(net)$type)]
V(net)$size <- ifelse(is.na(V(net)$Clade), 15, 10*sqrt(degree(net))) #5*sqrt(degree(net))
V(net)$label <- ifelse(is.na(V(net)$Clade), names(V(net)),  
                       unlist(lapply(strsplit(V(net)$Subtype, split="_"), "[", 1)))
V(net)$label <- str_replace_all(V(net)$label, c("alcicornis"="M.alc.", "annularis"="O.ann",
                                                "astreoides"="P.ast.", "cavernosa"="M.cav.",
                                                "cylindrus"="D.cyl.", "fragum"="F.fra.", 
                                                "furcata"="P.fur.", "radians"="S.rad.", 
                                                "siderea"="S.sid.", "strigosa"="D.str."))
V(net)$color <- ifelse(is.na(V(net)$Clade), "white",
                       taxcolors[factor(V(net)$Clade, levels=c("A","B","C","D","F","G"))])
E(net)$color <- "gray60"
E(net)$width <- 15 * E(net)$weight
set.seed(12347)
set.seed(12364)
set.seed(12374)
l <- layout.fruchterman.reingold(net)
l <- norm_coords(l, ymin=-1, ymax=1, xmin=-1, xmax=1)
par(mar=c(0,0,0,0))
plot(net, rescale=F, layout=l*1, edge.curved=0.1, vertex.label.cex=V(net)$size/20,
     vertex.label.color="black")
#legend(x=-1.05, y=1.1, c("CladeA", "CladeB", "CladeC", "CladeD", "CladeF", "CladeG"), pch=21,
#     col="#777777", pt.bg=taxcolors, pt.cex=1.1, cex=.6, bty="n", ncol=2, text.width=0.2)
```

## Dominant symbionts
#### (>50% relative abundance)
```{r dom_networks, fig.width=5, fig.height=5, fig.align="center"}
# Get network with one node for each species.
otudf <- data.frame(t(otu_table(phy97bs.f.p)), check.names=F)
# Percent of samples of species that contain OTU at more than 50% relative abundance
sp.ag <- aggregate(otudf, by=list(Species=data.frame(sample_data(phy97bs.f.p))$Species), 
                   FUN=function(x) length(x[x>0.5])/length(x))
rownames(sp.ag) <- sp.ag$Species
sp.ag <- sp.ag[, -1]
edges <- setNames(melt(as.matrix(sp.ag)), nm=c("source", "target", "weight"))  # Melt data
edges <- droplevels(edges[edges$weight>0, ])
# Create "nodes" table from sample and tax data
sdf <- data.frame(id=levels(data.frame(sample_data(phy97bs.f.p))$Species))
tdf <- data.frame(tax_table(phy97bs.f.p))[rownames(data.frame(tax_table(phy97bs.f.p))) %in% edges$target, ]
tdf$id <- rownames(tdf)
nodes <- merge(sdf, tdf, all=T)
nodes$type <- ifelse(is.na(nodes$Clade), 1, 2)
# Create network
net <- graph_from_data_frame(d=edges, vertices=nodes, directed=F)

# Best one for now ---------
V(net)$shape <- c("square", "circle")[factor(V(net)$type)]
V(net)$size <- ifelse(is.na(V(net)$Clade), 15, 10*(degree(net))^0.75) #5*sqrt(degree(net))
V(net)$label <- ifelse(is.na(V(net)$Clade), names(V(net)),  
                       unlist(lapply(strsplit(V(net)$Subtype, split="_"), "[", 1)))
V(net)$label <- str_replace_all(V(net)$label, c("alcicornis"="M.alc.", "annularis"="O.ann",
                                                "astreoides"="P.ast.", "cavernosa"="M.cav.",
                                                "cylindrus"="D.cyl.", "fragum"="F.fra.", 
                                                "furcata"="P.fur.", "radians"="S.rad.", 
                                                "siderea"="S.sid.", "strigosa"="D.str."))
V(net)$color <- ifelse(is.na(V(net)$Clade), "white",
                       taxcolors[factor(V(net)$Clade, levels=c("A","B","C","D","F","G"))])
E(net)$color <- "gray60"
E(net)$width <- 15 * E(net)$weight
set.seed(12347)
set.seed(12364)
set.seed(12374)
l <- layout.fruchterman.reingold(net)
l <- norm_coords(l, ymin=-1, ymax=1, xmin=-1, xmax=1)
par(mar=c(0,0,0,0))
plot(net, rescale=F, layout=l*1, edge.curved=0.1, vertex.label.cex=V(net)$size/20,
     vertex.label.color="black")
#legend(x=-1.05, y=1.1, c("CladeA", "CladeB", "CladeC", "CladeD", "CladeF", "CladeG"), pch=21,
#     col="#777777", pt.bg=taxcolors, pt.cex=1.1, cex=.6, bty="n", ncol=2, text.width=0.2)
```

## Background symbionts
#### (<1% relative abundance)
```{r bk.net, fig.width=5, fig.height=5, fig.align="center"}
# filter out OTUs that only occurred in one sample
phy97bs.f.p2 <- filter_taxa(phy97bs.f.p, function(x) sum(x>0) > 1, prune=TRUE)
# Get network with one node for each species.
otudf <- data.frame(t(otu_table(phy97bs.f.p2)), check.names=F)
# Whether any samples of species contain OTU at less than 1% relative abundance (but more than 0%)
sp.ag <- aggregate(otudf, by=list(Species=data.frame(sample_data(phy97bs.f.p2))$Species), 
                   FUN=function(x) any(x>0 & x<0.01))
rownames(sp.ag) <- sp.ag$Species
sp.ag <- sp.ag[, -1]
edges <- setNames(melt(as.matrix(sp.ag)), nm=c("source", "target", "weight"))  # Melt data
edges <- droplevels(edges[edges$weight>0, ])  # only keep if was observed in more than N colony of a spp
# Create "nodes" table from sample and tax data
sdf <- data.frame(id=levels(data.frame(sample_data(phy97bs.f.p2))$Species))
tdf <- data.frame(tax_table(phy97bs.f.p2))[rownames(data.frame(tax_table(phy97bs.f.p2))) %in% edges$target, ]
tdf$id <- rownames(tdf)
nodes <- merge(sdf, tdf, all=T)
nodes$type <- ifelse(is.na(nodes$Clade), 1, 2)
# Create network
net <- graph_from_data_frame(d=edges, vertices=nodes, directed=F)

# Edit vertex attributes
V(net)$shape <- c("square", "circle")[factor(V(net)$type)]
V(net)$size <- ifelse(is.na(V(net)$Clade), 15, 5*degree(net)) #5*sqrt(degree(net))
V(net)$label <- ifelse(is.na(V(net)$Clade), names(V(net)),  
                       unlist(lapply(strsplit(V(net)$Subtype, split="_"), "[", 1)))
V(net)$label <- str_replace_all(V(net)$label, c("alcicornis"="M.alc.", "annularis"="O.ann",
                                                "astreoides"="P.ast.", "cavernosa"="M.cav.",
                                                "cylindrus"="D.cyl.", "fragum"="F.fra.", 
                                                "furcata"="P.fur.", "radians"="S.rad.", 
                                                "siderea"="S.sid.", "strigosa"="D.str."))
V(net)$color <- ifelse(is.na(V(net)$Clade), "white",
                       taxcolors[factor(V(net)$Clade, levels=c("A","B","C","D","F","G"))])
E(net)$color <- "gray60"
E(net)$width <- E(net)$weight * 2

# Merge D nodes (assuming they are all D1a/trenchii)
# Get original node map (sequence starting at one to the total number of nodes)
nodemap <- seq(1:length(V(net)))
# Make a copy of the nodemap that will be modified
nodemapnew <- nodemap
# Find all nodes beginning with D, and give them the same node ID number
nodemapnew[grep("^D", V(net)$Subtype)] <- min(grep("^D", V(net)$Subtype))
# Replace the now missing values in the node map so node IDs remain sequential
while (max(setdiff(nodemap, nodemapnew)) < max(nodemap)) {
  nodemapnew[which(nodemapnew==max(setdiff(nodemap, nodemapnew))+1)] <- min(setdiff(nodemap, nodemapnew))
}
# Contract the network using the new nodemap (nodes with same node ID are merged)
net2 <- contract(net, nodemapnew, vertex.attr.comb=list(size="sum", "last"))
# Simplify network so that multiple edges are combined
net2 <- simplify(net2, remove.multiple=T, edge.attr.comb=list(weight="mean", width="mean", "ignore"))

# remove symbionts that are only connected to one or two coral species
net3 <- delete.vertices(net2, degree(net2)<=2) # & V(net2)$type==2)
V(net3)$size <- ifelse(is.na(V(net3)$Clade), 15, 1.5*degree(net3)^1.5)
set.seed(42384)
l3 <- layout.fruchterman.reingold(net3, weights=E(net3)$weight)
l3 <- norm_coords(l3, ymin=-1, ymax=1, xmin=-1, xmax=1)
par(mar=c(0,0,0,0))
plot(net3, rescale=F, layout=l3*1, edge.curved=0.1, vertex.label.cex=V(net3)$size/20,
     vertex.label.color="black")
points(0.9,0.9, pch=0, cex=4)
text(0.9,0.9, "O.ann.", cex=0.5)


set.seed(42399)
l3 <- layout_nicely(net3)
l3 <- norm_coords(l3, ymin=-1, ymax=1, xmin=-1, xmax=1)
par(mar=c(0,0,0,0))
V(net3)$size <- ifelse(is.na(V(net3)$Clade), 15, 10*degree(net3)^0.2)
plot(net3, rescale=F, layout=l3*1, edge.curved=0.1, vertex.label.cex=V(net3)$size/20,
     vertex.label.color="black")

```




# Beta diversity {.tabset}
- Beta diversity is evaluated here as the multivariate dispersion of samples within a coral species. Principal coordinate analysis on Bray-Curtis dissimilarities is used to calculate average distance-to-centroid values for each species group, which are then compared statistically by a permutation test. This analysis is implemented using betadisper() in the vegan package, based on [Anderson (2006)](dx.doi.org/10.1111/j.1541-0420.2005.00440.x).
- In this context, beta diveristy is analogous to 'flexibility' in symbiosis -- how different can corals of the same species be in terms of their symbiont communities? 
- The species with **highest beta diversity** are *D. strigosa*, *P. furcata*, and *M. alcicornis*, followed by *O. annularis* and *S. siderea*. Species with the **lowest beta diversity** are *F. fragum*, *D. cylindrus*, *S. radians*, *P. astreoides*, and *M. cavernosa*.

## 97% within-sample OTUs
```{r betad97bs, cache=T, fig.width=6, fig.height=6}
betad97bs <- betad(phy97bs.f.t, group="Species")
with(betad97bs$sambdsumm.ord, {
  plot(mean, type="n", main="97% within-sample OTUs",
       ylim=c(0, 0.75), ylab="Distance to centroid", xaxt="n", xlab="")
  arrows(1:10, mean - se, 1:10, mean + se, length=0.05, angle=90, code=3)
  points(1:10, mean, cex=1, pch=21, bg="white")
  text(1:10, par("usr")[3]-0.025, srt=45, adj=1, xpd=T, cex=0.75, 
       labels=levels(SAM$Species)[order(betad97bs$sambdsumm$mean, decreasing=T)])
  text(1:10, mean + se + 0.03, labels=betad97bs$saml$Letters[as.character(group)], cex=0.5)
})
```

## 100% OTUs
```{r betad100, cache=T, fig.width=6, fig.height=6}
betad100 <- betad(phy100.f.t, group="Species")
with(betad100$sambdsumm.ord, {
  plot(mean, type="n", main="100% OTUs",
       ylim=c(0, 0.75), ylab="Distance to centroid", xaxt="n", xlab="")
  arrows(1:10, mean - se, 1:10, mean + se, length=0.05, angle=90, code=3)
  points(1:10, mean, cex=1, pch=21, bg="white")
  text(1:10, par("usr")[3]-0.025, srt=45, adj=1, xpd=T, cex=0.75, 
       labels=levels(SAM$Species)[order(betad100$sambdsumm$mean, decreasing=T)])
  text(1:10, mean + se + 0.03, labels=betad100$saml$Letters[as.character(group)], cex=0.5)
})
```

## 97% OTUs
```{r betad97, cache=T, fig.width=6, fig.height=6}
betad97 <- betad(phy97.f.t, group="Species")
# Figure: Distance to centroids
with(betad97$sambdsumm.ord, {
  plot(mean, type="n", main="97% OTUs",
       ylim=c(0, 0.75), ylab="Distance to centroid", xaxt="n", xlab="")
  arrows(1:10, mean - se, 1:10, mean + se, length=0.05, angle=90, code=3)
  points(1:10, mean, cex=1, pch=21, bg="white")
  text(1:10, par("usr")[3]-0.025, srt=45, adj=1, xpd=T, cex=0.75, 
       labels=levels(SAM$Species)[order(betad97$sambdsumm$mean, decreasing=T)])
  text(1:10, mean + se + 0.03, labels=betad97$saml$Letters[as.character(group)], cex=0.5)
})
```

# Differences between species
Whether *Symbiodinium* communities differ among species is evaluated using permutational analysis of variance (PERMANOVA).

```{r permanovas, cache=T}
# PERMANOVA for difference among species
sppdiffs <- function(phy) {
  permanova.spp.t <- adonis(phyloseq::distance(phy, "bray") ~ Species, 
                          data=as(sample_data(phy), "data.frame"), permutations=9999)
  permanova.spp.t
  # Pairwise PERMANOVA tests for differences between species
  dmat <- as(phyloseq::distance(phy, "bray"), "matrix")
  df <- as(sample_data(phy), "data.frame")
  pairs <- data.frame(t(combn(levels(df$Species), 2)), stringsAsFactors=F)
  p.pairs <- setNames(rep(NA, nrow(pairs)), with(pairs, paste(X1, X2, sep='-')))
  for (i in 1:nrow(pairs)) {
    dfs <- subset(df, Species %in% c(pairs[i,1], pairs[i,2]))
    dmats <- dmat[rownames(dfs), rownames(dfs)]
    set.seed(152470)
    permanova <- adonis(dmats ~ Species, data=dfs, permutations=999)
    p.pairs[i] <- permanova$aov.tab$`Pr(>F)`[1]
  }
  # Return letters signifying differences 
  return(multcompLetters(p.pairs))
}

sppdiffs97 <- sppdiffs(phy97.f.t)
sppdiffs97bs <- sppdiffs(phy97bs.f.t)
sppdiffs100 <- sppdiffs(phy100.f.t)

sppdiffs <- data.frame("97% within-sample OTUs"=sppdiffs97bs$Letters,
                       "100% OTUs"=sppdiffs100$Letters, 
                       "97% OTUs"=sppdiffs97$Letters, check.names=F)
knitr::kable(sppdiffs, caption="Species not sharing a letter are significantly different (p < 0.05)")
```

##### Results:
- At the 97% within-sample OTU scale, *Symbiodinium* communities are **not significantly different** in the corals *M. alcicornis*, *O. annularis*, and *F. fragum*, as these corals are all dominantly associated with the same *Symbiodinium* OTU (BLAST identity=B1). These corals are **also not different** from *D. cylindrus* or *D. strigosa*, which also dominantly associated with the same B1 OTU, although the latter two species are different from each other. Meanwhile, *M. cavernosa* and *S. siderea* also share similar symbiont communities dominated by *Symbiodinium* C3. Finally, *S. radians*, *P. furcata*, and *P. astreoides* each have distinct, **significantly different** *Symbiodinium* communities.  
- At the 100% OTU scale, *Symbiodinium* communities were **significantly differentiated** in all species except for *O. annularis* and *D. strigosa* (sharing B1-dominated communities), and *S. siderea* and *M. cavernosa* (sharing C3-dominated communities).

# Differences between shores {.tabset}
Whether *Symbiodinium* communities differ between north vs. south shores within species is tested here using PERMANOVA.

## 97% within-sample OTUs
```{r shorediffs97bs, cache=T}
set.seed(43789)
set.seed(43789)
shorestats97bs <- perms(phy97bs.f.t, group="Species", trt="Shore")
knitr::kable(shorestats97bs, digits=3, row.names=F, 
             caption="97% within-sample OTU data: differences within and between shore for each species")
```

## 100% OTUs
```{r shorediffs100, cache=T}
set.seed(43789)
shorestats100 <- perms(phy100.f.t, group="Species", trt="Shore")
knitr::kable(shorestats100, digits=3, row.names=F, 
             caption="100% OTU data: differences within and between shore for each species")
```

## 97% OTUs
```{r shorediffs97, cache=T}
set.seed(43789)
shorestats97 <- perms(phy97.f.t, group="Species", trt="Shore")
knitr::kable(shorestats97, digits=3, row.names=F, 
             caption="97% OTU data: differences within and between shore for each species")
```

##### Results:
- __*P. furcata* is the only species whose symbiont communities are significantly different between the north and south shores__, at both the 97% within-sample OTU and 100% OTU scales. 


# Main findings
1. Symbiont communities are generally consistent with previous work on these species in the Caribbean, but high-throughput sequencing provides a much **more complete picture** of these communities’ composition.
2. South shore *P. furcata* are more likely to have clade A instead of C, and *S. siderea* more likely to have clade D instead of C.
3. **Within-sample OTU clustering** is more appropriate than clustering across samples.
4. We can apply **new types of analysis** with this much data, including network analysis and statistical tests on beta diversity.