---
title: "STJ2012 analysis"
author: "Ross Cunning"
date: "July 28, 2016"
output: 
  html_document: 
    toc: yes
---

```{r setup, include=FALSE}
knitr::opts_knit$set(root.dir = normalizePath(".."))
knitr::opts_chunk$set(echo = TRUE)

options(scipen = 3, digits = 2)
library(phyloseq); library(vegan); library(multcompView)

taxcolors <- matrix(c(rgb(141, 211, 199, maxColorValue=255), rgb(255, 255, 179, maxColorValue=255), 
                      rgb(190, 186, 218, maxColorValue=255), rgb(251, 128, 114, maxColorValue=255), 
                      rgb(253, 180, 98, maxColorValue=255)), 
                    dimnames=list(c("CladeA", "CladeB", "CladeC", "CladeD", "CladeF")))
```

# Overview
  This project investigates *Symbiodinium* communities in scleractinian corals sampled at various sites on the north and south shores of St. John, US Virgin Islands in August 2012. The ITS2 region of nrDNA isolated from coral samples was amplified and sequenced by paired-end 300bp reads on an Illumina MiSeq platform. Sequence data is processed through a custom bioinformatic pipeline and resulting OTU tables are analyzed by various multivariate statistical approaches to ask questions regarding alpha and beta diversity of *Symbiodinium* communities within and across coral species. The major aims of this work are to 1) characterize *Symbiodinium* communities in a range of coral species using high-throughput sequencing (many for the first time), and 2) to forward novel bioinformatic and statistical approaches to better understand *Symbiodinium* ecology. 

# Bioinformatics
  A [custom pipeline](sequence_processing.txt) is used here to process ITS2 sequence data. Briefly, paired reads were merged using the [illumina-utils](http://www.github.com/meren/illumina-utils) and filtered to keep only those with 3 mismatches or less. Chimeras were then removed by [usearch](http://www.drive5.com/usearch). From here, sequence data was processed using both of two approaches:
  
1. __97% OTU clustering within samples.__ With this approach, sequences from each individual sample were clustered at 97% similarity and the most abundant sequence from each cluster was selected as the representative sequence and assigned taxonomy by BLAST using a [custom ITS2 reference database](../data/Unaligned_ITS2_Database_23April13.fasta). Singletons were excluded. Identical OTUs across samples were then merged to create a global OTU table. By clustering at 97% _within_ samples, as opposed to across the entire dataset, OTUs from different samples that have _different representative sequences_ (i.e., different most abundant sequence within the 97% cluster) will remain separate even if these sequences are >97% similar to each other. This approach is used because _Symbiodinium_ ITS2 sequences are know to be multi-copy and vary intragenomically, and distinct _Symbiodinium_ types may contain identical sets of ITS2 sequences that occur in different relative proportions within the genome; therefore, selecting the most abundant sequence variant _within_ a sample prevents a potentially distinct _Symbiodinium_ type from being collapsed into an OTU with a different representative sequence that happens to be more abundant in other samples within the dataset.
2. __100% OTU clustering.__ All ITS2 sequences were clustered at 100% identity and assigned taxonomy by BLAST to the [custom reference database](../data/Unaligned_ITS2_Database_23April13.fasta). Singletons were excluded.
  
OTU tables and taxonomic assignments from both of these bioinformatic approaches are used in comparative downstream analyses.

# Data pre-processing
## Import dataset
The [phyloseq](https://joey711.github.io/phyloseq/) package is used here to combine the OTU table, taxonomic assignments, and sample metadata into a single R data object to facilitate downstream analyses.
```{r data_import, cache=T}
# Import OTU tables
OTU97 <- otu_table(read.table("data/97_otu_table.tsv", header=T, check.names=F, row.names=1,
                              skip=1, sep="\t", comment.char=""), taxa_are_rows=T)
OTU100 <- otu_table(read.table("data/100_otu_table.tsv", header=T, check.names=F, row.names=1,
                               skip=1, sep="\t", comment.char=""), taxa_are_rows=T)
# Import taxonomy
TAX97 <- tax_table(as.matrix(read.table("data/97_tax_assignments.txt", sep="\t", row.names=2,
                                        col.names=c("Clade", "OTU.ID", "Taxonomy", "Eval", "Subtype"))))
TAX100 <- tax_table(as.matrix(read.table("data/100_tax_assignments.txt", sep="\t", row.names=2,
                                         col.names=c("Clade", "OTU.ID", "Taxonomy", "Eval", "Subtype"))))
# Import sample data
SAM <- sample_data(read.delim("data/mapping_file.txt", sep="\t", header=T, row.names=1))
# Build phyloseq objects
phy97 <- phyloseq(OTU97, TAX97, SAM)
phy100 <- phyloseq(OTU100, TAX100, SAM)
```

#### Descriptive statistics and distribution of OTU counts
```{r phy_stats, fig.height=4, fig.width=8}
# Compute summary statistics
stats <- function(phy) {
  return(
    data.frame(rangeotus=paste0(range(taxa_sums(phy)), collapse=" - "),
               nsingles=length(taxa_sums(phy)[taxa_sums(phy) <= 1]),
               rangerps=paste0(range(sample_sums(phy)), collapse=" - "),
               amrps=paste(as.integer(mean(sample_sums(phy))), 
                           as.integer(sd(sample_sums(phy))), sep=" ± "),
               gmrps=paste(as.integer(exp(mean(log(sample_sums(phy))))),
                           as.integer(exp(sd(log(sample_sums(phy))))), sep=" ± "))
  )
}
stats97 <- stats(phy97)
stats100 <- stats(phy100)

# Create and plot histograms
phy <- phy97
taxhist97 <- hist(log10(taxa_sums(phy97)), plot=F)
samhist97 <- hist(log10(sample_sums(phy97)), plot=F)

taxhist100 <- hist(log10(taxa_sums(phy100)), plot=F)
samhist100 <- hist(log10(sample_sums(phy100)), plot=F)

par(mfrow=c(2, 2), mar=c(3,3,1,1))
plot(taxhist97, col="black", main="97% OTU counts", xlim=c(0, 5), las=1, mgp=c(2,0.5,0),
     xlab="No. sequences (log10)", ylab="No. OTUs", cex.lab=0.75, cex.axis=0.75)
plot(taxhist100, col="black", main="100% OTU counts", xlim=c(0, 5), las=1, mgp=c(2,0.5,0),
     xlab="No. sequences (log10)", ylab="No. OTUs", cex.lab=0.75, cex.axis=0.75)
plot(samhist97, col="black", main="97% OTU reads per sample", xlim=c(0, 5), las=1, mgp=c(2,0.5,0),
     xlab="No. sequences (log10)", ylab="No. samples", cex.lab=0.75, cex.axis=0.75)
plot(samhist100, col="black", main="100% OTU reads per sample", xlim=c(0, 5), las=1, mgp=c(2,0.5,0),
     xlab="No. sequences (log10)", ylab="No. samples", cex.lab=0.75, cex.axis=0.75)
```

|                                  |    97% OTUs               |    100% OTUs                |
| -------------------------------- | ------------------------- | --------------------------- |
| Total count in OTU table         | `r sum(otu_table(phy97))` |  `r sum(otu_table(phy100))` |
| Number of OTUs                   | `r ntaxa(phy97)`          |  `r ntaxa(phy100)`          |
| Range of OTU counts              | `r stats97$rangeotus`     | `r stats100$rangeotus`      |
| Number of singleton OTUs         | `r stats97$nsingles`      | `r stats100$nsingles`       |
| Number of samples                | `r nsamples(phy97)`       |  `r nsamples(phy100)`       |
| Range of reads per sample        | `r stats97$rangerps`      | `r stats100$rangerps`       |
| Mean (±SD) reads per sample (arithmetic) | `r stats97$amrps` | `r stats100$amrps`          |
| Mean (±SD) reads per sample (geometric)  | `r stats97$gmrps` | `r stats100$gmrps`          |

## Filter dataset
#### Filter samples by minimum count
```{r filter_samples}
# Set threshold number of reads
sn <- 10
# Remove samples with fewer reads than threshold
phy97.f <- prune_samples(sample_sums(phy97)>=sn, phy97)
phy100.f <- prune_samples(sample_sums(phy100)>=sn, phy100)
 
```

* Number of samples remaining in 97% OTU dataset: `r nsamples(phy97.f) `
* Number of samples remaining in 100% OTU dataset: `r nsamples(phy100.f) `

#### Filter OTUs by minimum count
```{r filter_taxa}
# Set threshold count
n <- 10
# Identify OTUs below threshold count
taxa97 <- taxa_sums(phy97.f)[which(taxa_sums(phy97.f) >= n)]  
taxa100 <- taxa_sums(phy100.f)[which(taxa_sums(phy100.f) >= n)]
# Remove taxa below threshold count
phy97.f <- prune_taxa(names(taxa97), phy97.f)  
phy100.f <- prune_taxa(names(taxa100), phy100.f)  
```

* Number of OTUs remaining in 97% dataset: `r ntaxa(phy97.f)`
* Number of OTUs remaining in 97% dataset: `r ntaxa(phy100.f)`


#### Filtered data: descriptive statistics and distribution of OTUs
```{r phy.f_histograms, fig.height=4, fig.width=8}
# Compute summary statistics
stats97.f <- stats(phy97.f)
stats100.f <- stats(phy100.f)

# Create and plot histograms
taxhist97 <- hist(log10(taxa_sums(phy97.f)), plot=F)
taxhist100 <- hist(log10(taxa_sums(phy100.f)), plot=F)
samhist97 <- hist(log10(sample_sums(phy97.f)), plot=F)
samhist100 <- hist(log10(sample_sums(phy100.f)), plot=F)

par(mfrow=c(2, 2), mar=c(3,3,1,1))
plot(taxhist97, col="black", main="97% OTU counts", xlim=c(0, 5), las=1, mgp=c(2,0.5,0),
     xlab="No. sequences (log10)", ylab="No. OTUs", cex.lab=0.75, cex.axis=0.75)
plot(taxhist100, col="black", main="100% OTU counts", xlim=c(0, 5), las=1, mgp=c(2,0.5,0),
     xlab="No. sequences (log10)", ylab="No. OTUs", cex.lab=0.75, cex.axis=0.75)
plot(samhist97, col="black", main="97% OTU reads per sample", xlim=c(0, 5), las=1, mgp=c(2,0.5,0),
     xlab="No. sequences (log10)", ylab="No. samples", cex.lab=0.75, cex.axis=0.75)
plot(samhist100, col="black", main="100% OTU reads per sample", xlim=c(0, 5), las=1, mgp=c(2,0.5,0),
     xlab="No. sequences (log10)", ylab="No. samples", cex.lab=0.75, cex.axis=0.75)
```

|                                  |    97% OTUs                     |    100% OTUs                |
| -------------------------------- | ------------------------------- | --------------------------- |
| Total count in OTU table         | `r sum(otu_table(phy97.f))`     | `r sum(otu_table(phy100.f))`|
| Number of OTUs                   | `r ntaxa(phy97.f)`              | `r ntaxa(phy100.f)`         |
| Range of OTU counts              | `r stats97.f$rangeotus`         | `r stats100.f$rangeotus`    |
| Number of singleton OTUs         | `r stats97.f$nsingles`          | `r stats100.f$nsingles`     |
| Number of samples                | `r nsamples(phy97.f)`           | `r nsamples(phy100.f)`      |
| Range of reads per sample        | `r stats97.f$rangerps`          | `r stats100.f$rangerps`     |
| Mean (±SD) reads per sample (arithmetic) | `r stats97.f$amrps`     | `r stats100.f$amrps`        |
| Mean (±SD) reads per sample (geometric)  | `r stats97.f$gmrps`     | `r stats100.f$gmrps`        |

## Transform count data
```{r transform}
# Convert to proportion (relative abundance)
phy97.f.p <- transform_sample_counts(phy97.f, function(x) x/sum(x))
phy100.f.p <- transform_sample_counts(phy100.f, function(x) x/sum(x))
# Apply transformation function
transform <- function(x) sqrt(x/sum(x))  # Set transformation function
phy97.f.t <- transform_sample_counts(phy97.f, transform)  # Transform data
phy100.f.t <- transform_sample_counts(phy100.f, transform)
```

-----

# Data analysis
## Community composition barplots
Here the composition of each sample is plotted as a horizontal bar, sorted by species and shore. Each segment of the bar represents an individual OTU. In the lefthand plot, all OTUs are colored by clade, in the right OTU, all OTUs are given a unique color.
```{r cladecomp97, cache=T, fig.height=6, fig.width=8}
# Define function to plot Symbiodinium community composition
composition <- function(phy, col, legend=T) {
  samdat <- data.frame(sample_data(phy))
  samdat$Genus <- factor(samdat$Genus, levels=rev(levels(samdat$Genus)))
  samdat$Species <- factor(samdat$Species, levels=rev(levels(samdat$Species)))
  samdat$Shore <- factor(samdat$Shore, levels=rev(levels(samdat$Shore)))
  samdat <- samdat[with(samdat, order(Genus, Species, Shore)), ]
  typerelabund <- as.matrix(otu_table(phy)[order(data.frame(tax_table(phy))$Subtype), 
                                           rownames(samdat)])
  shorebreaks <- c(as.character(samdat$Shore), "X")==c("X", as.character(samdat$Shore))
  shorebreaks <- which(shorebreaks==F) - 1
  spbreaks <- c(which(duplicated(samdat$Species)==F) - 1, nrow(samdat))
  # Make Barplot
  barplot(typerelabund, horiz=T, space=0, axes=F,axisnames=F, yaxs="i", col=col)
  rect(0, 0, par("usr")[2], par("usr")[4], lwd=1, xpd=T)
  axis(side=1, at=seq(0, 1, 0.1), line=0, tck=-0.025, mgp=c(0,0.25,0), cex.axis=0.7)
  mtext(side=1, "Relative abundance", cex=0.7, line=1)
  # Add legend
  if (legend==T) {
    legend(x=par("usr")[2]/2, y=par("usr")[4], xjust=0.5, yjust=0.25, horiz=T, bty="n", xpd=T, 
           cex=0.7, legend=c("A", "B", "C", "D", "F"), fill=taxcolors, x.intersp=0.5)
    legend(x=par("usr")[2]-0.03, y=par("usr")[4], xjust=0, yjust=0.1, bty="n", xpd=T, cex=0.6, 
           pt.cex=0.6, legend=c("N Shore", "S Shore"), fill=c(0, "gray40"), y.intersp=0.7, 
           x.intersp=0.3)
  }
  # Add grouping bars for Shore
  for (i in 1:length(shorebreaks)) {
    lines(c(0, 1), c(shorebreaks[i], shorebreaks[i]), lty=2, lwd=0.25)
    rect(1.01, shorebreaks[i], 1.04, shorebreaks[i+1], col=rep(c("gray40", 0), 50)[i], 
         lwd=0.25, xpd=T)
  }
  # Add lines to separate species and species names
  for (i in 1:length(spbreaks)) {
    lines(c(0, 1.07), c(spbreaks[i], spbreaks[i]), xpd=T, type="l", lwd=0.4)
    text(1.03, (spbreaks[i] + spbreaks[i+1]) / 2, xpd=T, pos=4, cex=0.6,
        labels=paste(samdat$Genus[which(duplicated(samdat$Species)==F)][i], "\n",
                    samdat$Species[which(duplicated(samdat$Species)==F)][i], sep=""))
  }
}

par(mfrow=c(1,2), mar=c(2, 1.5, 2, 5), lwd=0.1, cex=0.7)

# Plot composition of 97% OTUs colored by clade
composition(phy97.f.p, col=taxcolors[data.frame(tax_table(phy97.f.p))[order(data.frame(tax_table(phy97.f.p))$Subtype), ]$Clade], legend=T)

# Plot composition of 97% OTUs colored by OTU
composition(phy97.f.p, col=rainbow(ntaxa(phy97.f.p)), legend=F)

```

```{r cladecomp100, cache=T, fig.height=6, fig.width=8}
par(mfrow=c(1,2), mar=c(2, 1.5, 2, 5), lwd=0.1, cex=0.7)
# Plot composition of 100% OTUs colored by clade
composition(phy100.f.p, col=taxcolors[data.frame(tax_table(phy100.f.p))[order(data.frame(tax_table(phy100.f.p))$Subtype), ]$Clade], legend=T)

# Plot composition of 100% OTUs colored by OTU
composition(phy100.f.p, col=rainbow(ntaxa(phy100.f.p)), legend=F)
```

## Differences between species
Whether *Symbiodinium* communities differ among species is evaluated using permutational analysis of variance (PERMANOVA).

```{r permanovas, cache=T}
# PERMANOVA for difference among species
sppdiffs <- function(phy) {
  permanova.spp.t <- adonis(phyloseq::distance(phy, "bray") ~ Species, 
                          data=as(sample_data(phy), "data.frame"), permutations=9999)
  permanova.spp.t
  # Pairwise PERMANOVA tests for differences between species
  dmat <- as(phyloseq::distance(phy, "bray"), "matrix")
  df <- as(sample_data(phy), "data.frame")
  pairs <- data.frame(t(combn(levels(df$Species), 2)), stringsAsFactors=F)
  p.pairs <- setNames(rep(NA, nrow(pairs)), with(pairs, paste(X1, X2, sep='-')))
  for (i in 1:nrow(pairs)) {
    dfs <- subset(df, Species %in% c(pairs[i,1], pairs[i,2]))
    dmats <- dmat[rownames(dfs), rownames(dfs)]
    set.seed(152470)
    permanova <- adonis(dmats ~ Species, data=dfs, permutations=999)
    p.pairs[i] <- permanova$aov.tab$`Pr(>F)`[1]
  }
  # Return letters signifying differences 
  return(multcompLetters(p.pairs))
}

sppdiffs97 <- sppdiffs(phy97.f.t)
sppdiffs100 <- sppdiffs(phy100.f.t)

sppdiffs <- data.frame("97% OTUs"=sppdiffs97$Letters, "100% OTUs"=sppdiffs100$Letters, check.names=F)
knitr::kable(sppdiffs, caption="Species not sharing a letter are significantly different (p < 0.05)")
```

## Differences between shores
Whether *Symbiodinium* communities differ between north vs. south shores within species is tested here using PERMANOVA.

```{r shorediffs, cache=T}
shorediffs <- function(phy) {
  # Compute within- and between-shore statistics for each Species
  shorestats <- data.frame(matrix(dimnames=list(levels(data.frame(sample_data(phy))$Species), 
                                                c("Species", "n", "overall", "within", 
                                                  "between", "R2", "bd", "p")), ncol=8, nrow=10))
  for (i in 1:nlevels(data.frame(sample_data(phy))$Species)) {
    sp <<- levels(data.frame(sample_data(phy))$Species)[i]
    shorestats$Species[i] <- sp
    phy.species <- subset_samples(phy, Species==sp)  # Set temp phyloseq object
    shorestats$n[i] <- nsamples(phy.species)
    md <- meandist(phyloseq::distance(phy.species, "bray"), 
                   sample_data(phy.species)$Shore)
    shorestats$within[i] <- summary(md)$W  # Weighted mean dissimilarity within both Pools
    shorestats$between[i] <- summary(md)$B  # Mean dissimilarity between Pools
    shorestats$overall[i] <- summary(md)$D  # Overall dissimilarity
    if (nlevels(as(sample_data(phy.species), "data.frame")$Shore) > 1) {
      set.seed(70235)
      permanova <- adonis(phyloseq::distance(phy.species, "bray") ~ Shore, 
                          data=as(sample_data(phy.species), "data.frame"), permutations=999)
      shorestats$R2[i] <- permanova$aov.tab$"R2"[1]  # PERMANOVA partial R-squared
      shorestats$p[i] <- permanova$aov.tab$"Pr(>F)"[1]  # PERMANOVA p-value
      bd <- betadisper(phyloseq::distance(phy.species, "bray"),
                       group=as(sample_data(phy.species), "data.frame")$Shore)
      shorestats$bd[i] <- TukeyHSD(bd)$group[4]
    } else {
      shorestats$R2[i] <- NA
      shorestats$p[i] <- NA
      shorestats$bd[i] <- NA
    }
  }
  return(shorestats)
}

shorestats97 <- shorediffs(phy97.f.t)
shorestats100 <- shorediffs(phy100.f.t)

knitr::kable(shorestats97, row.names=F, caption="97% OTU data: differences within and between shore for each species")
knitr::kable(shorestats100, row.names=F, caption="100% OTU data: differences within and between shore for each species")
```

## Beta diversity
Beta diversity is evaluated here as the multivariate dispersion of samples within a coral species group. Principal coordinate analysis on Bray-Curtis dissimilarities is used to calculate average distance-to-centroid values for each species group, which are then compared statistically by a permutation test. This analysis is implemented using betadisper() in the vegan package, based on [Anderson (2006)](dx.doi.org/10.1111/j.1541-0420.2005.00440.x).
```{r betadisper, cache=T, fig.width=4, fig.height=4}
# Calculate distances to species centroids for each sample in principal coordinate space
sambd <- betadisper(d=vegdist(t(otu_table(phy97.f.t)), method="bray"), 
                    group=data.frame(sample_data(phy97.f.t))$Species, type="centroid",
                    bias.adjust=TRUE)
# Calculate each species' mean distance to centroid and standard error
sambdsumm <- aggregate(data.frame(mean=sambd$distances), by=list(group=sambd$group), FUN=mean)
sambdsumm$se <- aggregate(sambd$distances, by=list(group=sambd$group), 
                      FUN=function(x) sd(x)/sqrt(length(x)))$x
# Determine order of decreasing mean dispersion
sambdsumm.ord <- sambdsumm[order(sambdsumm$mean, decreasing=T), ]
# Update betadisper object with species in decreasing order of mean dispersion
sambd <- betadisper(d=vegdist(t(otu_table(phy97.f.t)), method="bray"), 
                    group=factor(data.frame(sample_data(phy97.f.t))$Species, levels=as.character(sambdsumm.ord$group)), type="centroid", bias.adjust=TRUE)
# Use permutations to perform pairwise comparisons of group mean dispersions
sampt <- permutest(sambd, pairwise=T)
saml <- multcompLetters(sampt$pairwise$permuted)

# Figure: Distance to centroids
par(mfrow=c(1,1), mar=c(3.5,4,1,1), lwd=1)
with(sambdsumm.ord, {
  plot(mean, type="n", ylim=c(0, 0.75), ylab="Distance to centroid", xaxt="n", xlab="")
  arrows(1:10, mean - se, 1:10, mean + se, length=0.05, angle=90, code=3)
  points(1:10, mean, cex=1, pch=21, bg="white")
  text(1:10, par("usr")[3]-0.025, srt=45, adj=1, xpd=T, cex=0.75, 
       labels=levels(SAM$Species)[order(sambdsumm$mean, decreasing=T)])
  text(1:10, mean + se + 0.03, labels=saml$Letters[as.character(group)], cex=0.5)
})
```

```{r strigosa}
dstr97 <- subset_samples(phy97.f.p, Species=="strigosa")
plot_bar(dstr97, x="Sample", y="Abundance", fill="OTU")
plot_bar(dstr97, x="Sample", y="Abundance", fill="Subtype")
# D strigosa hosted B1, B24, B35, B38, and C3 in Finney et al.... B25, B35, and B38 are missing from ref database.
# Also try 100% OTUs -- must filter more first...
dstr100.f <- subset_samples(phy100.f, Species=="strigosa")
dstr100.f.p <- transform_sample_counts(dstr100.f, function(x) x/sum(x))
#dstr100.f.p.f <- prune_taxa(rownames(otu_table(dstr100.f.p))[which(rowSums(otu_table(dstr100.f.p)) > 0.01)],
#                            dstr100.f.p)
#  
#plot_bar(dstr100.f.p.f, x="Sample", y="Abundance", fill="OTU")
#plot_bar(dstr100, x="Sample", y="Abundance", fill="Subtype")
#sample_names(dstr100.f)
#subset_samples(dstr100, Sample=="7")
#otu_table(dstr100.f)
#dstr100.f.p <- transform_sample_counts(dstr100.f, function(x) x/sum(x))
#colSums(otu_table(dstr100.f.p)[rowSums(otu_table(dstr100.f.p)) > 0.1, ])

```

```{r otu_barplots_custom}
# 100%-OTUs barplot 
par(mar=c(2,3,4,1))
samdat <- data.frame(sample_data(dstr100.f.p))
typerelabund <- as.matrix(otu_table(dstr100.f.p)[order(data.frame(tax_table(dstr100.f.p))$Subtype), 
                                                 rownames(samdat)])

# Get info for plotting OTU names and blast hits on top of barplot
blasthits <- as.character(data.frame(tax_table(dstr100.f.p))[order(data.frame(tax_table(dstr100.f.p))$Subtype)
                                                             , "Subtype"])
names <- as.character(rownames(data.frame(tax_table(dstr100.f.p)))[order(data.frame(tax_table(dstr100.f.p))$Subtype)])
divides <- apply(typerelabund, 2, cumsum)
heights <- diff(divides)
heights[heights < 0.04] <- NA

# Plot barplot and OTU names and blast hits
bars <- barplot(typerelabund, col=rainbow(941), border=NA, las=1, xlab="Sample", ylab="", cex.axis=0.75, cex.names=0.75)
mtext(side=2, text = "Relative Abundance", line=2)
for (i in 1:length(bars)) {
  text(rep(bars[i], length(heights[which(!is.na(heights[,i])),i])), 
       divides[which(!is.na(heights[,i])),i] + heights[which(!is.na(heights[,i])),i] / 2, 
       labels=paste(names[which(!is.na(heights[,i]))+1],blasthits[which(!is.na(heights[,i]))+1],sep="\n"),
       cex=0.6)
}

mtext(side=3, "B.) 100% OTUs", xpd=T, adj=0, line=2.5, font=2)

```

    